---
title: "its_16s_helianthus_statistical_analyses"
authors: "Michelle Odoi and Aaron Onufrak"
date: "June 2 2025"
output: rmdformats::downcute
editor_options: 
  chunk_output_type: console
---


# Code Overview

The following code describes the alpha and beta diversity analyses conducted on *Helianthus verticillatus*-associated microbiota. For the study the following plant-associated niches were sampled:

1. Flower buds (FB)
2. Leaves (LF)
3. Rhizomes (RZ)
4. Roots (RH)
5. Soil (SL)

Raw ITS2 and V4 sequences were processed into fungal and archaeal/bacterial ASVs and the following output files were generated: 

1. **seqtab_nochim_its_helianthus.csv**: Fungal ASV per sample abundances with chimeras removed.
2. **taxa_its_helianthus.csv**: Taxonomic assignments for fungal ASVs.
3. **seqtab.nochim.16s.csv**: Archaeal/bacterial ASV per sample abundances with chimeras removed.
4. **taxa.16s.csv**: Taxonomic assignments for archaeal/bacterial ASVs. 


# Loading Packages

```{r}
library(vegan)
library(dada2)
library(Biostrings)
library(ShortRead)
library(dplyr)
library(tidyr)
library(ggplot2)
library(car)
library(ape)
library(nlme)
library(emmeans)
library(ggpubr)
library(corrplot)
library(factoextra)
library(data.table)
library(multcompView)
library(hillR)
library(plyr)
library(corrplot)
library(readxl)
```

# Generating Functions

Creating custom functions to be used for statistical analyses. 

## Rarefaction Curve Plotter

The below function takes the following objects to generate rarefaction curves generated in ggplot:

1. **nonrarefied_asv_tab**: The original non-rarefied asv table 
2. **metadata**: The original metadata object that describes origins of samples
3. **vars**: Factor used to determine the cut-off for the maximum sequencing depth that can be used to retain at least 3 replicates per factor condition. 
4. **color**: metadata factor used to color lines

The function will return a list contatining the following objects:

1. rarefaction_obj: A dataframe containing the rarefaction sampling information for each sample
2. rarefaction_plot: A ggplot depicting the rarefaction curves. 

```{r}
# Constructing rarefaction curves for fungal root communities
rarefaction_plotter<-function(nonrarefied_asv_tab,metadata,vars,color,mode,depth=NULL){
  # Constructing rarefaction curves for fungal root communities
  rarefaction_tidy<-rarecurve(nonrarefied_asv_tab,tidy = TRUE)
  rarefaction_obj<-merge(metadata,rarefaction_tidy,by.x=0,by.y="Site")
  colnames(rarefaction_obj)<-sub("Row.names","Sample.ID",colnames(rarefaction_obj))
  replicate_min<-c()
  replicate_break<-c()
  for(i in 1:length(nonrarefied_asv_tab)){
    sample_depth<-sort(rowSums(nonrarefied_asv_tab))[i]
    rarefied_table<-suppressWarnings(as.data.frame(rrarefy(nonrarefied_asv_tab,sample=sample_depth)))
    ds.rarefied<-as.data.frame(subset(rarefied_table,rowSums(rarefied_table)>=sample_depth))
    test.df<-as.data.frame(merge(ds.rarefied,metadata,by.x=0,by.y=0)) %>%
  dplyr::count({{vars}},.drop=FALSE)
    print(test.df)
if(((nrow(test.df[test.df$n>=3,]))>0)&&(!(nrow(test.df[test.df$n<3,]))>=1)){
  replicate_min<-c(replicate_min,sample_depth)
}
if((nrow(test.df[test.df$n<3,]))>0){
  replicate_break<-c(replicate_break,sample_depth)
  break
}
    print(replicate_break)
    print(replicate_min[length(replicate_min)])
  }
  seq_counts_unrarefied<-sort(rowSums(nonrarefied_asv_tab))

  if(mode=="evaluation"){
  rarefaction_plot<-ggplot()+
  geom_line(data=rarefaction_obj,aes(x=Sample,y=Species,color=.data[[color]],group=.data[["Sample.ID"]]))+
  geom_vline(data=NULL,xintercept=c(seq_counts_unrarefied[c(1,round(length(seq_counts_unrarefied)*0.1),round(length(seq_counts_unrarefied)*0.30),round(length(seq_counts_unrarefied)*0.60),round(length(seq_counts_unrarefied)*0.9),length(seq_counts_unrarefied))]),linetype=2)+
  geom_vline(data=NULL,xintercept=replicate_min[length(replicate_min)],color="green")+
    geom_vline(data=NULL,xintercept=replicate_break[1],color="red")+
    scale_x_continuous(
      breaks=c(seq_counts_unrarefied[c(1,round(length(seq_counts_unrarefied)*0.1),round(length(seq_counts_unrarefied)*0.30),round(length(seq_counts_unrarefied)*0.60),round(length(seq_counts_unrarefied)*0.9),length(seq_counts_unrarefied))]),
      labels=c(seq_counts_unrarefied[c(1,round(length(seq_counts_unrarefied)*0.1),round(length(seq_counts_unrarefied)*0.3),round(length(seq_counts_unrarefied)*0.60),round(length(seq_counts_unrarefied)*0.9),length(seq_counts_unrarefied))]))+
  geom_text(data=NULL,aes(x=replicate_min[length(replicate_min)]-1000,y=max(rarefaction_obj$Species+10),label=replicate_min[length(replicate_min)],angle=90))+
      geom_text(data=NULL,aes(x=replicate_break[1]-1000,y=max(rarefaction_obj$Species+10),label=replicate_break[1],angle=90))+
    theme(panel.background = element_blank(),panel.border = element_rect(fill=NA,color="black"))+
  xlab("Number of Sequences")+
  ylab("ASV Count")
  rarefaction_information<-list(rarefaction_obj,rarefaction_plot)
  print(paste0("The maximum rarefaction cut-off that can be used to retain at least 3 replicates per treatment is ",replicate_min[length(replicate_min)]))
  }

 if(mode=="pub"){
  
  rarefaction_plot<-ggplot()+
  geom_line(data=rarefaction_obj,aes(x=Sample,y=Species,color=.data[[color]],group=.data[["Sample.ID"]]))+
  geom_vline(data=NULL,xintercept=depth,color="black")+
    theme(panel.background = element_blank(),panel.border = element_rect(fill=NA,color="black"))+
  xlab("Number of Sequences")+
  ylab("ASV Count")
  rarefaction_information<-list(rarefaction_obj,rarefaction_plot)
  print(paste0("The maximum rarefaction cut-off that can be used to retain at least 3 replicates per treatment is ",replicate_min[length(replicate_min)]))
}
  return(rarefaction_information)

    
}


```

## Rarefy Table Function

The below function takes a rarfied table as input and removes samples that do not have the minimum sampling depth and any ASV that has a sum across samples of zeros following rarefaction. The function takes as input the following items: 

1. **rarefied_table**: a rarefied ASV table with unrarefied samples and zero sum ASVs still present
2. **sample**: sequencing depth by which all samples were rarefied.

```{r}
# Removing unrarefied samples (those less than the minimum sequencing depth). Removing ASVs present in the ASV table that were only present in samples removed following rarefaction. These ASVs will have column sums of 0. 
rarefy_formatting<-function(nonrarefied_asv_tab,sample){
  rarefied_table<-suppressWarnings(as.data.frame(rrarefy(nonrarefied_asv_tab, sample=sample)))
  ds.rarefied<-as.data.frame(subset(rarefied_table, rowSums(rarefied_table)>=sample))
  ds.rare.asv<-ds.rarefied[,colSums(ds.rarefied)>0]
  if(all(rowSums(ds.rare.asv)==sample)){
    print(paste0("All samples rarefied to ",sample))
  }
  return(ds.rare.asv)
}
```

## Function to Calculate Hill Numbers and store in a table

```{r}
hill_table<-function(rare_asv_table){
q0<-data.frame(q0=hill_taxa(select_if(rare_asv_table,is.numeric),q=0))
q1<-data.frame(q1=hill_taxa(select_if(rare_asv_table,is.numeric),q=1))
q2<-data.frame(q2=hill_taxa(select_if(rare_asv_table,is.numeric),q=2))
hill.table.met<-cbind(select_if(rare_asv_table,is.character),q0,q1,q2)
return(hill.table.met)}
```

## PCoA and Ordination Function

The below function wraps the vegdist function from vegan and the pcoa function from ape and generates an ordination using ggplot. The results of the PCoA and the ggplot are stored in a two object list. The first is the PCoA output and the second is the ggplot that can be further customized using ggplot arguments. The function requires the following input:

1. **table**: a rarefied ASV table
2. **method**: a method for computing distances (see: ??vegan::vegdist for options)
3. **point_ellipse_color**: The variable by which points and ellipses should be colored
4. **shape**: Shaping variable
5. **polygon**: Logical value indiciating whether filled ellipses should be plotted to reprsented. This is useful when working with data with multiple factors. **Must be specified with ellipse_fill has a value.** 
6. **ellipse_fill**: Variable by which filled ellipses should be colored. 
7. **point_size**: Size of points for ordination. Default is 3.5

```{r}
pcoa_imager<-function(table,method,point_ellipse_color,shape=NULL,polygon=NULL,ellipse_fill=NULL,point_size=3.5){
  column_grabber<-`[[` # Generating a subsetting function
  point_ellipse_color<-column_grabber(table,point_ellipse_color) # will subset the column that contains the provided variable by which points and ellipses should be colored
  ellipse_fill<-column_grabber(table,ellipse_fill) # will subset the column that contains the variabe by which ellipses will be filled
  shape<-column_grabber(table,shape) # will subset the column that contains the point shaping variabe 
  relabund.asv<-decostand(Filter(x=table,f=is.numeric), method="total") # using the decostand function from vegan to convert count data to relative abundances
  dist.asv<-vegdist(relabund.asv, method=method) # calculating distance matrix using the vegdist function from vegan. Method is a user specified variable to indicate which method for computing distance to be used. 
  pcoa.asv<-pcoa(dist.asv) # using the pcoa function from ape to perform a PCoA with the distance matrix stored in dist.asv. 
  pcoavec.asv<-as.data.frame(pcoa.asv$vectors) # creating a dataframe that only includes the vectors from the PCoA.
  pcoasitescores.asv<-data.frame(PC1=pcoavec.asv$Axis.1, PC2=pcoavec.asv$Axis.2) # creating a datatable that is only the first and second PCs. 
  
  # creating a if/else statement based on presence of ellipse_fill. 
      if (is.null(ellipse_fill)){
      pcoagraph.asv<-data.frame(pcoasitescores.asv,PC1=pcoasitescores.asv$PC1, PC2=pcoasitescores.asv$PC2, color=point_ellipse_color, shape_var=shape)
    } else {
  pcoagraph.asv<-data.frame(pcoasitescores.asv,PC1=pcoasitescores.asv$PC1, PC2=pcoasitescores.asv$PC2, color=point_ellipse_color, ellipse_fill=ellipse_fill, shape_var=shape)}
  
  # Generates standard deviation ellipses based on color variable
  pcoaellipse<-ordiellipse(pcoasitescores.asv,pcoagraph.asv$color, display="sites", kind="sd", draw="none")
  ell <- data.frame()
for(g in levels(as.factor(pcoagraph.asv$color))){
ell <- rbind(ell, cbind(as.data.frame(with(pcoagraph.asv[pcoagraph.asv$color==g,],                                                vegan:::veganCovEllipse(pcoaellipse[[g]]$cov,pcoaellipse[[g]]$center,pcoaellipse[[g]]$scale))) ,color=g))}
  
  # Generates secondary ellipse for second color
pcoaellipse.2<-ordiellipse(pcoasitescores.asv,pcoagraph.asv$ellipse_fill, display="sites", kind="sd", draw="none")
  ell_2 <- data.frame()
for(g in levels(as.factor(pcoagraph.asv$ellipse_fill))){
ell_2 <- rbind(ell_2, cbind(as.data.frame(with(pcoagraph.asv[pcoagraph.asv$elllipse_fill==g,],                                                vegan:::veganCovEllipse(pcoaellipse.2[[g]]$cov,pcoaellipse.2[[g]]$center,pcoaellipse.2[[g]]$scale))) ,ellipse_fill=g))}  
  
# Plotting first two PCs of PCoA in ggplot  
  
  # If an ellipse_fill variable is present polygon must be set to true
  if (polygon==TRUE){
  # If statement that indicates what to do in the absence of a shape variable
    if (is.null(shape)){
      pcoa_plot<-ggplot(pcoagraph.asv, aes(PC1,PC2, colour=color))+
      geom_point(size=point_size)+
      geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  geom_polygon(data=ell_2, aes(x=PC1, y=PC2, fill=ellipse_fill),linewidth=0.5, linetype=2,inherit.aes = FALSE,alpha=0.1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))
   
       } else {
  pcoa_plot<-ggplot(pcoagraph.asv, aes(PC1,PC2, colour=color))+       
  geom_point(aes(shape=shape_var),size=point_size)+
  geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  geom_polygon(data=ell_2, aes(x=PC1, y=PC2, fill=ellipse_fill),linewidth=0.5, linetype=2,inherit.aes = FALSE,alpha=0.1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))}
         
    } else {
      
      if (is.null(shape)){
      pcoa_plot<-ggplot(pcoagraph.asv, aes(PC1,PC2, colour=color))+
      geom_point(size=point_size)+
      geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", linewidth =1, fill=NA),
    panel.background = element_rect(fill="white"))+
    xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))
      
        } else {
 pcoa_plot<-ggplot(pcoagraph.asv, aes(PC1,PC2, colour=color))+ 
  geom_point(aes(shape=shape_var), size=point_size)+
  geom_path(data=ell, aes(x=PC1, y=PC2, colour=color),linewidth=0.5, linetype=1)+
 theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA),
    panel.background = element_rect(fill="white"))+
      xlab(paste0("PC1"," (",round(column_grabber(pcoa.asv,"values")[1,2]*100,2),"%)"))+
    ylab(paste0("PC2"," (",round(column_grabber(pcoa.asv,"values")[2,2]*100,2),"%)"))}
    }
  pcoa_func_out<-list(pcoa.asv,pcoa_plot)
  return(pcoa_func_out)
}

```

## Taxanomy formatting

```{r}
taxonomy_formatting<-function(rare.asvtab,taxonomytab){
  t.rareasv<-as.data.frame(t(rare.asvtab))
filtered.tax<-as.data.frame(subset(taxonomytab,row.names(taxonomytab) %in% row.names(t.rareasv)))
  t.rareasv.tax<-merge(t.rareasv,filtered.tax, by="row.names")
  return(t.rareasv.tax)
}
```

## Taxonomy String Cleaner

A function to clean-up taxonomy string names so that they are more aesthetically pleasing. Function takes as input the following object:

**taxonomy.table**: A taxonomy table object. 

```{r}
taxonomy_string_clean<-function(taxonomy.table){
    taxonomy.table[["Kingdom"]]<-sub("k__","",taxonomy.table[["Kingdom"]])
    taxonomy.table[["Phylum"]]<-sub("p__","",taxonomy.table[["Phylum"]])
    taxonomy.table[["Class"]]<-sub("c__","",taxonomy.table[["Class"]])
    taxonomy.table[["Order"]]<-sub("o__","",taxonomy.table[["Order"]])
    taxonomy.table[["Family"]]<-sub("f__","",taxonomy.table[["Family"]])
    taxonomy.table[["Genus"]]<-sub("g__","",taxonomy.table[["Genus"]])
    taxonomy.table[["Species"]]<-sub("s__","",taxonomy.table[["Species"]])
    taxonomy.table[is.na(taxonomy.table)]<-"Unassigned"
    return(taxonomy.table)
  }
```


# Data Import and Formatting

```{r}
# Importing fungal taxa
taxa_its_helianthus<-fread("taxa_its_helianthus.csv",header=TRUE,sep=',')

# Importing fungal ASV table
seqtab_nochim_its_helianthus<-fread("seqtab_nochim_its_helianthus.csv",header=TRUE,sep=',')

# Sample Metadata ITS
met_helianthus_its<-read.table("Cultureindependent_metadata_its.csv",header=TRUE,sep=',')
row.names(met_helianthus_its)<-met_helianthus_its$Sample_ID


# Importing archaeal/bacterial taxa
taxa_16s_helianthus<-fread("taxa.16s.csv",header=TRUE,sep=',')

# Importing archaeal/bacterial ASV table
seqtab_nochim_16s_helianthus<-fread("seqtab.nochim.16s.csv",header=TRUE,sep=',')

# Sample Metadata 16s
met_helianthus_16s<-read.table("Cultureindependent_metadata_16s.csv",header=TRUE,sep=',')
row.names(met_helianthus_16s)<-met_helianthus_16s$Sample_ID
```

## Taxonomy filtering and data frame formatting for fungi

```{r}
# Number of taxa prior to taxonomy filtering.
nrow(taxa_its_helianthus)

# Converting taxonomy table to a data frame
taxa_original_its_helianthus<-as.data.frame(taxa_its_helianthus)

# Make ASV sequence the row name
row.names(taxa_original_its_helianthus)<-taxa_original_its_helianthus$V1

# Removing the duplicate column (ASV Sequence or V1)
taxa_original_its_helianthus$V1<-NULL

# Removing sequences not identified at the phylum level
# This line will remove sequences not identified to at least phylum ie NA at the phylum level 
taxa_na_omit_its_helianthus<-taxa_original_its_helianthus[-(which(is.na(taxa_original_its_helianthus$Phylum))),] 

# Checking number of ASVs lost
nrow(taxa_original_its_helianthus)-nrow(taxa_na_omit_its_helianthus)

# Determining number of ASVs retained after taxonomy filtering
nrow(taxa_na_omit_its_helianthus)

# Converting ASV table to a data frame
seqtab_nochim_its_helianthus<-as.data.frame(seqtab_nochim_its_helianthus)

# Making sample names the row names
row.names(seqtab_nochim_its_helianthus)<-seqtab_nochim_its_helianthus$V1

# Using sub to remove "filtered_" from the beginning part of the sample name from the sequence table
row.names(seqtab_nochim_its_helianthus)<-sub("filtered_","",row.names(seqtab_nochim_its_helianthus))

# Removing file extensions from end of sample names
row.names(seqtab_nochim_its_helianthus)<-sub(".r1.fq.gz","",row.names(seqtab_nochim_its_helianthus))

# Removing duplicate column at V1
seqtab_nochim_its_helianthus$V1<-NULL

# Confirming row names
row.names(seqtab_nochim_its_helianthus)

# Checking number of sequences per sample
rowSums(seqtab_nochim_its_helianthus)

# Transposing ASV Table to allow for merging of ASV table and taxonomy
t_seqtab_nochim_its_helianthus<-as.data.frame(t(seqtab_nochim_its_helianthus))

# Merging ASV and taxonomy tables
t_seqtab_nochim_its_filt_helianthus<-t_seqtab_nochim_its_helianthus[row.names(t_seqtab_nochim_its_helianthus)%in%row.names(taxa_na_omit_its_helianthus),]

# Removing one of the samples that was sequenced twice (FDHRZP3). Removing sample from the Pilot study. FSBRZP3 was not sequenced.  
t_seqtab_nochim_its_filt_helianthus$Hv_DH_RZ<-NULL

# Checking to ensure that the column was removed
colnames(t_seqtab_nochim_its_filt_helianthus)

# Changing colnames from samples in the pilot study to match metadata. 
colnames(t_seqtab_nochim_its_filt_helianthus)[151] <- "FRTLFP4" 
colnames(t_seqtab_nochim_its_filt_helianthus)[152] <-  "FRTRHP1" 
colnames(t_seqtab_nochim_its_filt_helianthus)[153] <- "FSBFBP1"

# Correcting sample name from FSFSLP8 to FSBSLP8
colnames(t_seqtab_nochim_its_filt_helianthus)[150] <- "FSBSLP8"

# Removing ASvs that are present in the negative controls
t_seqtab_nochim_its_filt_no_zero_helianthus<-t_seqtab_nochim_its_filt_helianthus[rowSums(t_seqtab_nochim_its_filt_helianthus[,grep("Negative",colnames(t_seqtab_nochim_its_filt_helianthus))])==0,]

# Checking that no ASVs are left in the negative controls
colSums(t_seqtab_nochim_its_filt_no_zero_helianthus[,grep("Negative",colnames(t_seqtab_nochim_its_filt_no_zero_helianthus))])

# Checking number of sequences per sample
sort(colSums(t_seqtab_nochim_its_filt_no_zero_helianthus))

# Checking total number of ASVs
nrow(t_seqtab_nochim_its_filt_no_zero_helianthus) 

sum(t_seqtab_nochim_its_filt_no_zero_helianthus)

# Merging taxonomy information into ASV table after removing the negative ASV from all samples
t_seqtab_nochim_its_filt_no_zero_helianthus<-merge(t_seqtab_nochim_its_filt_no_zero_helianthus,taxa_na_omit_its_helianthus, by="row.names")

# Checking that taxonomy information is present
colnames(t_seqtab_nochim_its_filt_no_zero_helianthus)

#  Creating ASV labels and make these new row names
asvnumber_its_helianthus<-as.character(c(1:nrow(t_seqtab_nochim_its_filt_no_zero_helianthus)))
asvnumber_its_helianthus<-paste0("asv_its.",labels(asvnumber_its_helianthus))

# Creating new row names that are the ASV labels opposed to the ASV sequence
row.names(t_seqtab_nochim_its_filt_no_zero_helianthus)<-asvnumber_its_helianthus

# Pulling out soil samples since they will be analysed separately
t_seqtab_nochim_its_filt_no_zero_helianthus_soil<-t_seqtab_nochim_its_filt_no_zero_helianthus[,grep("SL|Kingdom|Phylum|Class|Order|Family|Genus|Species",colnames(t_seqtab_nochim_its_filt_no_zero_helianthus))]

# Checking that only soils were selected
colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_soil)

# Pulling out plant samples (ie leaves, flower buds, roots and rhizomes) since they will be analysed separately
t_seqtab_nochim_its_filt_no_zero_helianthus_plant<-t_seqtab_nochim_its_filt_no_zero_helianthus[,grep("Row.names|SL|Negative",colnames(t_seqtab_nochim_its_filt_no_zero_helianthus), invert=TRUE)]

# Checking that only plant niches wre selected
colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_plant)

# Splitting taxonomy information from ASV table for soils
t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_soil <- t_seqtab_nochim_its_filt_no_zero_helianthus_soil[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=TRUE, colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_soil))]

# Sanity check to make sure file contains only soil
colnames(t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_soil)

# Taxonomy information for only soil ASVs
taxonomy_nochim_its_filt_no_negative_soil<-t_seqtab_nochim_its_filt_no_zero_helianthus_soil[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=FALSE,colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_soil))]

# Sanity check to make sure only taxonomy info is stored
colnames(taxonomy_nochim_its_filt_no_negative_soil)

# Splitting taxonomy information from ASV table for plant samples
t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_plant <- t_seqtab_nochim_its_filt_no_zero_helianthus_plant[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=TRUE, colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_plant))]

# Sanity check to make sure file contains only plant samples
colnames(t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_plant)

# Taxonomy info for only plant ASVs
taxonomy_nochim_its_filt_no_negative_plant<-t_seqtab_nochim_its_filt_no_zero_helianthus_plant[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=FALSE,colnames(t_seqtab_nochim_its_filt_no_zero_helianthus_plant))]

# Sanity check to make sure only taxonomy information is stored. 
colnames(taxonomy_nochim_its_filt_no_negative_plant)

```

## Taxonomy filtering and data frame formatting for archaea/bacteria

```{r}
# Number of taxa prior to taxonomy filtering.
nrow(taxa_16s_helianthus)

# Converting taxonomy table to a data frame
taxa_original_16s_helianthus<-as.data.frame(taxa_16s_helianthus)

# Make ASV sequence the row name
row.names(taxa_original_16s_helianthus)<-taxa_original_16s_helianthus$V1

# Removing the duplicate column (ASV Sequence or V1)
taxa_original_16s_helianthus$V1<-NULL

# Removing sequences not identified at the phylum level
taxa_na_omit_16s_helianthus<-taxa_original_16s_helianthus[-(which(is.na(taxa_original_16s_helianthus$Phylum))),] 

# Removing chloroplast sequences
taxa_chloroplast_omit_16s_helianthus<-taxa_na_omit_16s_helianthus[grep("Chloroplast",taxa_na_omit_16s_helianthus$Order,invert=TRUE),]

# Removing eukaryote sequences
taxa_euk_omit_16s_helianthus<-taxa_chloroplast_omit_16s_helianthus[grep("Eukaryota",taxa_chloroplast_omit_16s_helianthus$Kingdom,invert=TRUE),]

# Removing mitochondria sequences #63,848
taxa_mitochondria_omit_16s_helianthus<-taxa_euk_omit_16s_helianthus[grep("Mitochondria",taxa_euk_omit_16s_helianthus$Family,invert=TRUE),]

# Checking number of ASVs lost
nrow(taxa_original_16s_helianthus)-nrow(taxa_mitochondria_omit_16s_helianthus)

# Determining number of ASVs retained after taxonomy filtering
nrow(taxa_mitochondria_omit_16s_helianthus)

# Converting ASV table to a data frame
seqtab_nochim_16s_helianthus<-as.data.frame(seqtab_nochim_16s_helianthus)

# Making sample names the row names
row.names(seqtab_nochim_16s_helianthus)<-seqtab_nochim_16s_helianthus$V1

# Using sub to remove "filtered_" from the beginning part of the sample name from the sequence table
row.names(seqtab_nochim_16s_helianthus)<-sub("filtered_","",row.names(seqtab_nochim_16s_helianthus))

# Removing file extensions from end of sample names
row.names(seqtab_nochim_16s_helianthus)<-sub(".r1.fq.gz","",row.names(seqtab_nochim_16s_helianthus))

# Removing duplicate column at V1
seqtab_nochim_16s_helianthus$V1<-NULL

# Confirming row names
row.names(seqtab_nochim_16s_helianthus)

# Checking number of sequences per sample
rowSums(seqtab_nochim_16s_helianthus)

# Transposing ASV Table to allow for merging of ASV table and taxonomy
t_seqtab_nochim_16s_helianthus<-as.data.frame(t(seqtab_nochim_16s_helianthus))

# Merging ASV and taxonomy tables
t_seqtab_nochim_16s_filt_helianthus<-t_seqtab_nochim_16s_helianthus[row.names(t_seqtab_nochim_16s_helianthus)%in%row.names(taxa_mitochondria_omit_16s_helianthus),]

# Removing one of the samples that was sequenced twice (BDHRZP3). Removing sample from the Pilot study. BSBRZP3 was not sequenced.  

t_seqtab_nochim_16s_filt_helianthus$DH_RZ_P3_b.raw_1.fq.gz<-NULL

# Checking to ensure that the column was removed
colnames(t_seqtab_nochim_16s_filt_helianthus)

# Changing colnames from samples in the pilot study to match metadata. 
colnames(t_seqtab_nochim_16s_filt_helianthus)[151] <- "BRTLFP4" 
colnames(t_seqtab_nochim_16s_filt_helianthus)[152] <-  "BRTRHP1" 
colnames(t_seqtab_nochim_16s_filt_helianthus)[153] <- "BSBFBP1"

# Removing ASvs that are present in the negative controls
t_seqtab_nochim_16s_filt_no_zero_helianthus<-t_seqtab_nochim_16s_filt_helianthus[rowSums(t_seqtab_nochim_16s_filt_helianthus[,grep("Negative",colnames(t_seqtab_nochim_16s_filt_helianthus))])==0,]

# Checking that no ASVs are left in the negative controls
colSums(t_seqtab_nochim_16s_filt_no_zero_helianthus[,grep("Negative",colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus))])

# Checking number of sequences per sample
sort(colSums(t_seqtab_nochim_16s_filt_no_zero_helianthus))

# Checking total number of ASVs
nrow(t_seqtab_nochim_16s_filt_no_zero_helianthus) 

sum(t_seqtab_nochim_16s_filt_no_zero_helianthus)

# Merging taxonomy information into ASV table after removing the negative ASV from all samples
t_seqtab_nochim_16s_filt_no_zero_helianthus<-merge(t_seqtab_nochim_16s_filt_no_zero_helianthus,taxa_mitochondria_omit_16s_helianthus, by="row.names")

# Checking that taxonomy information is present
colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus)

#  Creating ASV labels and make these new row names
asvnumber_16s_helianthus<-as.character(c(1:nrow(t_seqtab_nochim_16s_filt_no_zero_helianthus)))
asvnumber_16s_helianthus<-paste0("asv_16s.",labels(asvnumber_16s_helianthus))

# Creating new row names that are the ASV labels opposed to the ASV sequence
row.names(t_seqtab_nochim_16s_filt_no_zero_helianthus)<-asvnumber_16s_helianthus

# Pulling out soil samples since they will be analysed separately
t_seqtab_nochim_16s_filt_no_zero_helianthus_soil<-t_seqtab_nochim_16s_filt_no_zero_helianthus[,grep("SL|Kingdom|Phylum|Class|Order|Family|Genus|Species",colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus))]

# Pulling out plant samples (ie leaves, flower buds, roots and rhizomes) since they will be analysed separately
t_seqtab_nochim_16s_filt_no_zero_helianthus_plant<-t_seqtab_nochim_16s_filt_no_zero_helianthus[,grep("Row.names|SL|Negative",colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus), invert=TRUE)]

# Splitting taxonomy information from ASV table for soils
t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_soil <- t_seqtab_nochim_16s_filt_no_zero_helianthus_soil[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=TRUE, colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus_soil))]

# Sanity check to make sure file contains only soil
colnames(t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_soil)

# Taxonomy information for only soil ASVs
taxonomy_nochim_16s_filt_no_negative_soil<-t_seqtab_nochim_16s_filt_no_zero_helianthus_soil[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=FALSE,colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus_soil))]

# Splitting taxonomy information from ASV table for plant samples
t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_plant <- t_seqtab_nochim_16s_filt_no_zero_helianthus_plant[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=TRUE, colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus_plant))]

# Sanity check to make sure file contains only plant samples
colnames(t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_plant)

# Taxonomy info for only plant ASVs
taxonomy_nochim_16s_filt_no_negative_plant<-t_seqtab_nochim_16s_filt_no_zero_helianthus_plant[,grep("Kingdom|Phylum|Class|Order|Family|Genus|Species",invert=FALSE,colnames(t_seqtab_nochim_16s_filt_no_zero_helianthus_plant))]

```


# Sample Rarefaction

## Rarefying Fungal Communities

### All Plant Niches

```{r}
set.seed(123)
# Transposing plant ASV data frame to be used in vegan
asv_tab_its_helianthus_plant<-t(t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_plant)

# Sanity check for only flower buds
nrow(asv_tab_its_helianthus_plant)
sort(rowSums(asv_tab_its_helianthus_plant))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_plant_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_plant)))
asv_tab_its_helianthus_plant_counts

# Generating rarefaction plot
rarefaction_its_plant<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_plant, metadata=met_helianthus_its,vars=Tissue_Type,color="Tissue_Type",mode = "pub", depth=6058)

# Pulling out rarefaction curves
rareplot.its_plant<-rarefaction_its_plant[[2]]
rareplot.its_plant

# Rarefying the ITS plant
rareasv_its_plant<-rarefy_formatting(asv_tab_its_helianthus_plant,sample=6058)

```

### Flower Buds Only

```{r}
# Using grep to subset and pull out only flower bud samples
asv_tab_its_helianthus_flower<-asv_tab_its_helianthus_plant[grep("FB",rownames(asv_tab_its_helianthus_plant)),]

# Sanity check for only flower buds
nrow(asv_tab_its_helianthus_flower)
sort(rowSums(asv_tab_its_helianthus_flower))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_flower_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_flower)))
asv_tab_its_helianthus_flower_counts

# Generating rarefaction plot
rarefaction_its_fb<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_flower, metadata=met_helianthus_its,vars=Tissue_Type,color="Site",mode = "pub", depth=4129)

# Pulling out rarefaction curves
rareplot.its_fb<-rarefaction_its_fb[[2]]
rareplot.its_fb

# Rarefying the ITS flower bud table
rareasv_its_flower<-rarefy_formatting(asv_tab_its_helianthus_flower,sample=4129)

# Sanity check
rowSums(rareasv_its_flower)

```


### Leaves Only

```{r}
# Using grep to subset and pull out only leaf samples
asv_tab_its_helianthus_leaf<-asv_tab_its_helianthus_plant[grep("LF",rownames(asv_tab_its_helianthus_plant)),]

# Sanity check for only leaves
nrow(asv_tab_its_helianthus_leaf)
sort(rowSums(asv_tab_its_helianthus_leaf))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_leaf_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_leaf)))
asv_tab_its_helianthus_leaf_counts

# Generating rarefaction plot
rarefaction_its_leaf<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_leaf, metadata=met_helianthus_its,vars=Tissue_Type,color="Site",mode = "pub", depth=19458)

# Pulling out rarefaction curves
rareplot.its_leaf<-rarefaction_its_leaf[[2]]
rareplot.its_leaf

# Rarefying the ITS leaf ASV table 
rareasv_its_leaf<-rarefy_formatting(asv_tab_its_helianthus_leaf,sample=19458)

# Sanity check
rowSums(rareasv_its_leaf)
```

### Roots Only

```{r}
# Using grep to subset and pull out only root samples
asv_tab_its_helianthus_root<-asv_tab_its_helianthus_plant[grep("RH",rownames(asv_tab_its_helianthus_plant)),]

# Sanity check for only roots
nrow(asv_tab_its_helianthus_root)
sort(rowSums(asv_tab_its_helianthus_root))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_root_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_root)))
asv_tab_its_helianthus_root_counts

# Generating rarefaction plot
rarefaction_its_root<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_root, metadata=met_helianthus_its,vars=Tissue_Type,color="Site",mode = "pub", depth=25747)

# Pulling out rarefaction curves
rareplot.its_root<-rarefaction_its_root[[2]]
rareplot.its_root

# Rarefying the ITS roots ASV table 
rareasv_its_root<-rarefy_formatting(asv_tab_its_helianthus_root,sample=25747)


# Sanity check
rowSums(rareasv_its_root)
```

### Rhizome Only

```{r}
# Using grep to subset and pull out only rhizome samples
asv_tab_its_helianthus_rhizome<-asv_tab_its_helianthus_plant[grep("RZ",rownames(asv_tab_its_helianthus_plant)),]

# Sanity check for only rhizome
nrow(asv_tab_its_helianthus_rhizome)
sort(rowSums(asv_tab_its_helianthus_rhizome))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_rhizome_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_rhizome)))
asv_tab_its_helianthus_rhizome_counts

# Generating rarefaction plot
rarefaction_its_rhizome<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_rhizome, metadata=met_helianthus_its,vars=Tissue_Type,color="Site",mode = "pub", depth=7338)

# Pulling out rarefaction curves
rareplot.its_rhizome<-rarefaction_its_rhizome[[2]]
rareplot.its_rhizome

# Rarefying the ITS rhizome ASV table 
rareasv_its_rhizome<-rarefy_formatting(asv_tab_its_helianthus_rhizome,sample=7338)

```

### Soil Only

```{r}
# Using grep to subset and pull out only soil samples
asv_tab_its_helianthus_soil<-as.data.frame(t(t_seqtab_nochim_its_filt_no_zero_no_tax_helianthus_soil))
# Sanity check for only soil
nrow(asv_tab_its_helianthus_soil)
sort(rowSums(asv_tab_its_helianthus_soil))

# Determining number of sequences per sample prior to rarefaction
asv_tab_its_helianthus_soil_counts<-as.data.frame(sort(rowSums(asv_tab_its_helianthus_soil)))
asv_tab_its_helianthus_soil_counts

# Generating rarefaction plot
rarefaction_its_soil<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_its_helianthus_soil, metadata=met_helianthus_its,vars=Tissue_Type,color="Site",mode = "pub", depth=56910)

# Pulling out rarefaction curves
rareplot.its_soil<-rarefaction_its_soil[[2]]
rareplot.its_soil

# Rarefying the ITS soil ASV table 
rareasv_its_soil<-rarefy_formatting(asv_tab_its_helianthus_soil,sample=56910)

#Organizing rarefaction cut-offs 

its_rarefaction_helianthus<-ggarrange(rareplot.its_plant,rareplot.its_fb,rareplot.its_leaf,rareplot.its_rhizome,rareplot.its_root,rareplot.its_soil, nrow=3, ncol = 2, common.legend=F, labels = c("A","B","C","D","E", "F"), vjust=1.0,widths=1, font.label = list(size = 15))
ggsave("its_rarefaction_helianthus.png", plot = its_rarefaction_helianthus, dpi = 300, width = 10, height = 10)
```


## Rarefying Bacterial Communities

### All Plant Niches

```{r}
set.seed(123)
# Transposing plant ASV data frame to be used in vegan
asv_tab_16s_helianthus_plant<-t(t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_plant)

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_plant_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_plant)))
asv_tab_16s_helianthus_plant_counts

# Generating rarefaction plot
rarefaction_16s_plant<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_plant, metadata=met_helianthus_16s,vars=Tissue_Type,color="Tissue_Type",mode = "pub", depth=1645)

# Pulling out rarefaction curves
rareplot.16s_plant<-rarefaction_16s_plant[[2]]
rareplot.16s_plant

# Rarefying the 16s plant
rareasv_16s_plant<-rarefy_formatting(asv_tab_16s_helianthus_plant,sample=1645)

# Sanity Check
rowSums(rareasv_16s_plant)
```

### Flower Buds Only

```{r}
# Using grep to subset and pull out only flower bud samples
asv_tab_16s_helianthus_flower<-asv_tab_16s_helianthus_plant[grep("FB",rownames(asv_tab_16s_helianthus_plant)),]

# Sanity check for only flower buds
nrow(asv_tab_16s_helianthus_flower)
sort(rowSums(asv_tab_16s_helianthus_flower))

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_flower_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_flower)))
asv_tab_16s_helianthus_flower_counts

# Generating rarefaction plot
rarefaction_16s_fb<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_flower, metadata=met_helianthus_16s,vars=Tissue_Type,color="Site",mode = "pub", depth=1645)

# Pulling out rarefaction curves
rareplot.16s_fb<-rarefaction_16s_fb[[2]]
rareplot.16s_fb

# Rarefying the 16s flower bud table
rareasv_16s_flower<-rarefy_formatting(asv_tab_16s_helianthus_flower,sample=1645)

# Sanity Check
rowSums(rareasv_16s_flower)
```


### Leaves Only

```{r}
# Using grep to subset and pull out only leaf samples
asv_tab_16s_helianthus_leaf<-asv_tab_16s_helianthus_plant[grep("LF",rownames(asv_tab_16s_helianthus_plant)),]

# Sanity check for only leaves
nrow(asv_tab_16s_helianthus_leaf)
sort(rowSums(asv_tab_16s_helianthus_leaf))

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_leaf_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_leaf)))
asv_tab_16s_helianthus_leaf_counts

# Generating rarefaction plot
rarefaction_16s_leaf<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_leaf, metadata=met_helianthus_16s,vars=Tissue_Type,color="Site",mode = "pub", depth=1961)

# Pulling out rarefaction curves
rareplot.16s_leaf<-rarefaction_16s_leaf[[2]]
rareplot.16s_leaf

# Rarefying the 16s leaf ASV table 
rareasv_16s_leaf<-rarefy_formatting(asv_tab_16s_helianthus_leaf,sample=1961)

# Sanity Check
rowSums(rareasv_16s_leaf)
```

### Roots Only

```{r}
# Using grep to subset and pull out only root samples
asv_tab_16s_helianthus_root<-asv_tab_16s_helianthus_plant[grep("RH",rownames(asv_tab_16s_helianthus_plant)),]

# Sanity check for only roots
nrow(asv_tab_16s_helianthus_root)
sort(rowSums(asv_tab_16s_helianthus_root))

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_root_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_root)))
asv_tab_16s_helianthus_root_counts

# Generating rarefaction plot
rarefaction_16s_root<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_root, metadata=met_helianthus_16s,vars=Tissue_Type,color="Site",mode = "pub", depth=8071)

# Pulling out rarefaction curves
rareplot.16s_root<-rarefaction_16s_root[[2]]
rareplot.16s_root

# Rarefying the 16s roots ASV table 
rareasv_16s_root<-rarefy_formatting(asv_tab_16s_helianthus_root,sample=8071)

# Sanity Check
rowSums(rareasv_16s_root)
```

### Rhizosphere Only

```{r}
# Using grep to subset and pull out only rhizome samples
asv_tab_16s_helianthus_rhizome<-asv_tab_16s_helianthus_plant[grep("RZ",rownames(asv_tab_16s_helianthus_plant)),]

# Sanity check for only rhizome
nrow(asv_tab_16s_helianthus_rhizome)
sort(rowSums(asv_tab_16s_helianthus_rhizome))

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_rhizome_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_rhizome)))
asv_tab_16s_helianthus_rhizome_counts

# Generating rarefaction plot
rarefaction_16s_rhizome<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_rhizome, metadata=met_helianthus_16s,vars=Tissue_Type,color="Site",mode = "pub", depth=5095)

# Pulling out rarefaction curves
rareplot.16s_rhizome<-rarefaction_16s_rhizome[[2]]
rareplot.16s_rhizome

# Rarefying the 16s rhizome ASV table 
rareasv_16s_rhizome<-rarefy_formatting(asv_tab_16s_helianthus_rhizome,sample=5095)

# Sanity Check
rowSums(rareasv_16s_rhizome)
```

### Soil Only

```{r}
# Using grep to subset and pull out only soil samples
asv_tab_16s_helianthus_soil<-as.data.frame(t(t_seqtab_nochim_16s_filt_no_zero_no_tax_helianthus_soil))

# Sanity check for only soil
nrow(asv_tab_16s_helianthus_soil)
sort(rowSums(asv_tab_16s_helianthus_soil))

# Determining number of sequences per sample prior to rarefaction
asv_tab_16s_helianthus_soil_counts<-as.data.frame(sort(rowSums(asv_tab_16s_helianthus_soil)))
asv_tab_16s_helianthus_soil_counts

# Generating rarefaction plot
rarefaction_16s_soil<-rarefaction_plotter(nonrarefied_asv_tab =asv_tab_16s_helianthus_soil, metadata=met_helianthus_16s,vars=Tissue_Type,color="Site",mode = "pub", depth=16520)

# Pulling out rarefaction curves
rareplot.16s_soil<-rarefaction_16s_soil[[2]]
rareplot.16s_soil

# Rarefying the 16s soil ASV table 
rareasv_16s_soil<-rarefy_formatting(asv_tab_16s_helianthus_soil,sample=16520)

# Sanity Check
rowSums(rareasv_16s_soil)

#Organizing rarefaction cut-offs 
bac_rarefaction_helianthus<-ggarrange(rareplot.16s_plant,rareplot.16s_fb,rareplot.16s_leaf,rareplot.16s_rhizome,rareplot.16s_root,rareplot.16s_soil, nrow=3, ncol = 2, common.legend=F, labels = c("A","B","C","D","E", "F"), vjust=1.0,widths=1, font.label = list(size = 15))
ggsave("bac_rarefaction_helianthus.png", plot = bac_rarefaction_helianthus, dpi = 300, width = 10, height = 10)

```

# Merging Metadata into rarefied ASV tables

## Fungi

```{r}
# All Plant Tissues
met.rareasv.its_plant<-merge(met_helianthus_its,rareasv_its_plant,by.x=0, by.y=0)


# Removing duplicate row name
met.rareasv.its_plant$Row.names<-NULL

# Flower buds
met.rareasv.its_flower<-merge(met_helianthus_its,rareasv_its_flower,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.its_flower$Row.names<-NULL

# Leaf
met.rareasv.its_leaf<-merge(met_helianthus_its,rareasv_its_leaf,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.its_leaf$Row.names<-NULL

# Rhizome
met.rareasv.its_rhizome<-merge(met_helianthus_its,rareasv_its_rhizome,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.its_rhizome$Row.names<-NULL

# Root
met.rareasv.its_root<-merge(met_helianthus_its,rareasv_its_root,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.its_root$Row.names<-NULL

# Soil
met.rareasv.its_soil<-merge(met_helianthus_its,rareasv_its_soil,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.its_soil$Row.names<-NULL

```

## Bacteria
```{r}
# All Plant Tissues
met.rareasv.16s_plant<-merge(met_helianthus_16s,rareasv_16s_plant,by.x=0, by.y=0)


# Removing duplicate row name
met.rareasv.16s_plant$Row.names<-NULL

# Flower buds
met.rareasv.16s_flower<-merge(met_helianthus_16s,rareasv_16s_flower,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.16s_flower$Row.names<-NULL

# Leaf
met.rareasv.16s_leaf<-merge(met_helianthus_16s,rareasv_16s_leaf,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.16s_leaf$Row.names<-NULL

# Rhizome
met.rareasv.16s_rhizome<-merge(met_helianthus_16s,rareasv_16s_rhizome,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.16s_rhizome$Row.names<-NULL

# Root
met.rareasv.16s_root<-merge(met_helianthus_16s,rareasv_16s_root,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.16s_root$Row.names<-NULL

# Soil
met.rareasv.16s_soil<-merge(met_helianthus_16s,rareasv_16s_soil,by.x=0, by.y=0)

# Removing duplicate row name
met.rareasv.16s_soil$Row.names<-NULL

```


# Alpha-Diversity Analyses

## Fungi

First calculating alpha-diversity for each niche. Using the hill_table function made above to calculate each hill index and store in a table in one go. 

```{r}
# Whole Plant ITS
hill.its_plant<-hill_table(met.rareasv.its_plant)

# Flower Buds
hill.its_flower<-hill_table(met.rareasv.its_flower)

# Leaf
hill.its_leaf<-hill_table(met.rareasv.its_leaf)

# Root
hill.its_root<-hill_table(met.rareasv.its_root)

# Rhizome
hill.its_rhizome<-hill_table(met.rareasv.its_rhizome)

# Soil
hill.its_soil<-hill_table(met.rareasv.its_soil)
```

### Parametric Tests

First exploring how well the data fits the assumptions of paramteric models. 

```{r}
# Performing an ANOVA with the aov function and assessing how well it fits model assumptions. 

# All Niches


# q0
# Model meets parametric assumptions
hill.its_plant_aov_q0<-aov((q0)~Site*Tissue_Type,hill.its_plant)
par(mfrow=c(2,2))
plot(hill.its_plant_aov_q0)
Anova(hill.its_plant_aov_q0,type="III")
TukeyHSD(hill.its_plant_aov_q0)

# Interaction term not significant so dropping and re-running model. 
hill.its_plant_aov_q0<-aov((q0)~Site+Tissue_Type,hill.its_plant)
par(mfrow=c(2,2))
plot(hill.its_plant_aov_q0)
Anova(hill.its_plant_aov_q0,type="III")
TukeyHSD(hill.its_plant_aov_q0)


# q1
# Model meets assumptions when square root transformed
hill.its_plant_aov_q1<-aov(sqrt(q1)~Site*Tissue_Type,hill.its_plant)
par(mfrow=c(2,2))
plot(hill.its_plant_aov_q1)
Anova(hill.its_plant_aov_q1,type="III")
TukeyHSD(hill.its_plant_aov_q1)


# q2
# Transformations do not improve model fit. Switching to non-parametric alernative. 
hill.its_plant_aov_q2<-aov((q2)~Site*Tissue_Type,hill.its_plant)
par(mfrow=c(2,2))
plot(hill.its_plant_aov_q2)
Anova(hill.its_plant_aov_q2,type="III")

```

### Non-Parametric Tests
Because the data failed to meet the assumptions of the parametric tests for q2 , will use the non-parametric kruskal-wallis test alternative for all hill values. 

```{r}
# Using a Kruskal-Wallis Test with a post-hoc dunn test. 

# All Niches by Site q0
hill.its_plant_kruskal_q0<-kruskal.test(q0~Site,hill.its_plant)
hill.its_plant_kruskal_q0
rstatix::dunn_test(hill.its_plant,q0~Site)


# Plotting results in ggplot
hill.its_plant_q0_site_plot<-ggplot(hill.its_plant,aes(x=Site,y=q0,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q0 (ASV Richness)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=810,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=810,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=810,label="B"))
hill.its_plant_q0_site_plot

# All niches by site q1
hill.its_plant_kruskal_q1<-kruskal.test(q1~Site,hill.its_plant)
hill.its_plant_kruskal_q1
rstatix::dunn_test(hill.its_plant,q1~Site)

# Plotting results in ggplot
hill.its_plant_q1_site_plot<-ggplot(hill.its_plant,aes(x=Site,y=q1,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=300,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=300,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=300,label="B"))
hill.its_plant_q1_site_plot

# All niches by site q2
hill.its_plant_kruskal_q2<-kruskal.test(q2~Site,hill.its_plant)
hill.its_plant_kruskal_q2
rstatix::dunn_test(hill.its_plant,q2~Site)

# Plotting results in ggplot
hill.its_plant_q2_site_plot<-ggplot(hill.its_plant,aes(x=Site,y=q2,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=150,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=150,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=150,label="B"))
hill.its_plant_q2_site_plot


# All Niches by Niche q0
hill.its_plant_kruskal_niche_q0<-kruskal.test(q0~Tissue_Type,hill.its_plant)
hill.its_plant_kruskal_niche_q0
rstatix::dunn_test(hill.its_plant,q0~Tissue_Type)

# Plotting results in ggplot
hill.its_plant_q0_niche_plot<-ggplot(hill.its_plant,aes(x=Tissue_Type,y=q0,fill=Tissue_Type))+
  labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q0 (ASV Richness)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=810,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=810,label="BC"))+
  geom_text(data=NULL,size=6,aes(x=3,y=810,label="AC"))+
  geom_text(data=NULL,size=6,aes(x=4,y=810,label="B"))
hill.its_plant_q0_niche_plot

# All Niches by Niche q1
hill.its_plant_kruskal_niche_q1<-kruskal.test(q1~Tissue_Type,hill.its_plant)
hill.its_plant_kruskal_niche_q1

# Plotting results in ggplot
hill.its_plant_q1_niche_plot<-ggplot(hill.its_plant,aes(x=Tissue_Type,y=q1,fill=Tissue_Type))+
  labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.its_plant_q1_niche_plot

# All Niches by Niche q2
hill.its_plant_kruskal_niche_q2<-kruskal.test(q2~Tissue_Type,hill.its_plant)
hill.its_plant_kruskal_niche_q2

# Plotting results in ggplot
hill.its_plant_q2_niche_plot<-ggplot(hill.its_plant,aes(x=Tissue_Type,y=q2,fill=Tissue_Type))+
  labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.its_plant_q2_niche_plot

its_hill_all<-ggarrange(hill.its_plant_q0_site_plot, hill.its_plant_q0_niche_plot, hill.its_plant_q1_site_plot, hill.its_plant_q1_niche_plot, hill.its_plant_q2_site_plot, hill.its_plant_q2_niche_plot, nrow=3,ncol=2,common.legend =F, labels=c("A","B","C", "D", "E", "F"),font.label = list(size = 15))
                        
ggsave("its_hill_all.png", plot = its_hill_all, dpi = 300, width = 15, height = 11)   

# Soils by Site q0
hill.its_soil_kruskal_q0<-kruskal.test(q0~Site,hill.its_soil)
hill.its_soil_kruskal_q0

# Plotting results in ggplot
hill.its_soil_q0_site_plot<-ggplot(hill.its_soil,aes(x=Site,y=q0,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q0 (ASV Richness)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.its_soil_q0_site_plot

# All niches by site q1
hill.its_soil_kruskal_q1<-kruskal.test(q1~Site,hill.its_soil)
hill.its_soil_kruskal_q1
rstatix::dunn_test(hill.its_soil,q1~Site)

# Plotting results in ggplot
hill.its_soil_q1_site_plot<-ggplot(hill.its_soil,aes(x=Site,y=q1,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=400,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=400,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=400,label="B"))
hill.its_soil_q1_site_plot

# All niches by site q2
hill.its_soil_kruskal_q2<-kruskal.test(q2~Site,hill.its_soil)
hill.its_soil_kruskal_q2

# Plotting results in ggplot
hill.its_soil_q2_site_plot<-ggplot(hill.its_soil,aes(x=Site,y=q2,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.its_soil_q2_site_plot

                    
                      
its_hill_soil<-ggarrange(hill.its_soil_q0_site_plot,hill.its_soil_q1_site_plot,hill.its_soil_q2_site_plot, labels=c("A","B","C"),common.legend=T,legend="right", vjust=1.0,widths=1, ncol = 3, font.label = list(size = 15))

ggsave("its_hill_soil.png", plot = its_hill_soil, dpi = 300, width = 10, height = 5)  

```


## Bacteria

```{r}
# Whole Plant 16s
hill.16s_plant<-hill_table(met.rareasv.16s_plant)


# Flower Buds
hill.16s_flower<-hill_table(met.rareasv.16s_flower)

# Leaf
hill.16s_leaf<-hill_table(met.rareasv.16s_leaf)

# Root
hill.16s_root<-hill_table(met.rareasv.16s_root)

# Rhizome
hill.16s_rhizome<-hill_table(met.rareasv.16s_rhizome)

# Soil
hill.16s_soil<-hill_table(met.rareasv.16s_soil)
```

### Non-Parametric Tests
Because the data failed to meet the assumptions of the parametric tests for fungi, will use the non-parametric kruskal-wallis test alternative for all hill values for bacteria. 

```{r}
# Using a Kruskal-Wallis Test with a post-hoc dunn test. 

# All Niches by Site q0
hill.16s_plant_kruskal_q0<-kruskal.test(q0~Site,hill.16s_plant)
hill.16s_plant_kruskal_q0


# Plotting results in ggplot
hill.16s_plant_q0_site_plot<-ggplot(hill.16s_plant,aes(x=Site,y=q0,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q0 (ASV Richness)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.16s_plant_q0_site_plot

# All niches by site q1
hill.16s_plant_kruskal_q1<-kruskal.test(q1~Site,hill.16s_plant)
hill.16s_plant_kruskal_q1
rstatix::dunn_test(hill.16s_plant,q1~Site)

# Plotting results in ggplot
hill.16s_plant_q1_site_plot<-ggplot(hill.16s_plant,aes(x=Site,y=q1,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.16s_plant_q1_site_plot

# All niches by site q2
hill.16s_plant_kruskal_q2<-kruskal.test(q2~Site,hill.16s_plant)
hill.16s_plant_kruskal_q2
rstatix::dunn_test(hill.16s_plant,q2~Site)

# Plotting results in ggplot
hill.16s_plant_q2_site_plot<-ggplot(hill.16s_plant,aes(x=Site,y=q2,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.16s_plant_q2_site_plot


 # All Niches by Niche q0
hill.16s_plant_kruskal_niche_q0<-kruskal.test(q0~Tissue_Type,hill.16s_plant)
hill.16s_plant_kruskal_niche_q0
rstatix::dunn_test(hill.16s_plant,q0~Tissue_Type)

# Plotting results in ggplot
hill.16s_plant_q0_niche_plot<-ggplot(hill.16s_plant,aes(x=Tissue_Type,y=q0,fill=Tissue_Type))+
   labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q0 (ASV Richness)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=1150,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=1150,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=1150,label="B"))+
  geom_text(data=NULL,size=6,aes(x=4,y=1150,label="B"))
hill.16s_plant_q0_niche_plot

# All Niches by Niche q1
hill.16s_plant_kruskal_niche_q1<-kruskal.test(q1~Tissue_Type,hill.16s_plant)
hill.16s_plant_kruskal_niche_q1
rstatix::dunn_test(hill.16s_plant,q1~Tissue_Type)

# Plotting results in ggplot
hill.16s_plant_q1_niche_plot<-ggplot(hill.16s_plant,aes(x=Tissue_Type,y=q1,fill=Tissue_Type))+
   labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
    geom_text(data=NULL,size=6,aes(x=1,y=1000,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=1000,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=1000,label="B"))+
  geom_text(data=NULL,size=6,aes(x=4,y=1000,label="B"))
hill.16s_plant_q1_niche_plot

# All Niches by Niche q2
hill.16s_plant_kruskal_niche_q2<-kruskal.test(q2~Tissue_Type,hill.16s_plant)
hill.16s_plant_kruskal_niche_q2
rstatix::dunn_test(hill.16s_plant,q2~Tissue_Type)

# Plotting results in ggplot
hill.16s_plant_q2_niche_plot<-ggplot(hill.16s_plant,aes(x=Tissue_Type,y=q2,fill=Tissue_Type))+
   labs(x = "Niche", fill = "Niche")+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Niche")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=1000,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=2,y=1000,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=1000,label="BC"))+
  geom_text(data=NULL,size=6,aes(x=4,y=1000,label="C"))
hill.16s_plant_q2_niche_plot

bac_hill_all<-ggarrange(hill.16s_plant_q0_site_plot, hill.16s_plant_q0_niche_plot, hill.16s_plant_q1_site_plot, hill.16s_plant_q1_niche_plot, hill.16s_plant_q2_site_plot, hill.16s_plant_q2_niche_plot, nrow=3,ncol=2,common.legend =F, labels=c("A","B","C", "D", "E", "F"),font.label = list(size = 15))
                        
ggsave("bac_hill_all.png", plot = bac_hill_all, dpi = 300, width = 15, height = 11)   

# Soils by Site q0
hill.16s_soil_kruskal_q0<-kruskal.test(q0~Site,hill.16s_soil)
hill.16s_soil_kruskal_q0
rstatix::dunn_test(hill.16s_soil,q0~Site)

# Plotting results in ggplot
hill.16s_soil_q0_site_plot<-ggplot(hill.16s_soil,aes(x=Site,y=q0,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q0 (ASV Richness")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
    geom_text(data=NULL,size=6,aes(x=1,y=3850,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=2,y=3850,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=3850,label="B"))
hill.16s_soil_q0_site_plot

# All niches by site q1
hill.16s_soil_kruskal_q1<-kruskal.test(q1~Site,hill.16s_soil)
hill.16s_soil_kruskal_q1
rstatix::dunn_test(hill.16s_soil,q1~Site)

# Plotting results in ggplot
hill.16s_soil_q1_site_plot<-ggplot(hill.16s_soil,aes(x=Site,y=q1,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q1 (Shannon Diversity)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=2500,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=2,y=2500,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=2500,label="B"))
hill.16s_soil_q1_site_plot

# All niches by site q2
hill.16s_soil_kruskal_q2<-kruskal.test(q2~Site,hill.16s_soil)
hill.16s_soil_kruskal_q2
rstatix::dunn_test(hill.16s_soil,q2~Site)

# Plotting results in ggplot
hill.16s_soil_q2_site_plot<-ggplot(hill.16s_soil,aes(x=Site,y=q2,fill=Site))+
  geom_boxplot(outlier.shape=NA)+
  geom_point(position=position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Hill q2 (Inverse Simpson)")+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))
hill.16s_soil_q2_site_plot


bac_hill_soil<-ggarrange(hill.16s_soil_q0_site_plot,hill.16s_soil_q1_site_plot,hill.16s_soil_q2_site_plot, labels=c("A","B","C"),common.legend=T,legend="right", vjust=1.0,widths=1, ncol = 3, font.label = list(size = 15))

ggsave("bac_hill_soil.png", plot = bac_hill_soil, dpi = 300, width = 10, height = 5)

        
```

# Beta-Diversity Analyses

## Fungi

### Principal Coordinate Analyses

```{r}
set.seed(123)
# All Niches

# Generating PCoA and ordination plot
pcoa_its_all_niche<-pcoa_imager(table = met.rareasv.its_plant,method = "bray",point_ellipse_color = "Tissue_Type",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_all_niche_plot<-pcoa_its_all_niche[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_all_niche_plot

# Conducting PERMANOVA
permanova_its_all_niche<-as_tibble(adonis2(select_if(met.rareasv.its_plant,is.numeric)~met.rareasv.its_plant$Site*met.rareasv.its_plant$Tissue_Type,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_all_niche

# Flower Buds

# Generating PCoA and ordination plot
pcoa_its_flower<-pcoa_imager(table = met.rareasv.its_flower,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_flower_plot<-pcoa_its_flower[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_flower_plot

# Conducting PERMANOVA
permanova_its_flower<-as_tibble(adonis2(select_if(met.rareasv.its_flower,is.numeric)~met.rareasv.its_flower$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_flower

# Leaf

# Generating PCoA and ordination plot
pcoa_its_leaf<-pcoa_imager(table = met.rareasv.its_leaf,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_leaf_plot<-pcoa_its_leaf[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_leaf_plot

# Conducting PERMANOVA
permanova_its_leaf<-as_tibble(adonis2(select_if(met.rareasv.its_leaf,is.numeric)~met.rareasv.its_leaf$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_leaf


# Rhizome

# Generating PCoA and ordination plot
pcoa_its_rhizome<-pcoa_imager(table = met.rareasv.its_rhizome,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_rhizome_plot<-pcoa_its_rhizome[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_rhizome_plot

# Conducting PERMANOVA
permanova_its_rhizome<-as_tibble(adonis2(select_if(met.rareasv.its_rhizome,is.numeric)~met.rareasv.its_rhizome$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_rhizome

# Roots 

# Generating PCoA and ordination plot
pcoa_its_root<-pcoa_imager(table = met.rareasv.its_root,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_root_plot<-pcoa_its_root[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_root_plot

# Conducting PERMANOVA
permanova_its_root<-as_tibble(adonis2(select_if(met.rareasv.its_root,is.numeric)~met.rareasv.its_root$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_root

#Organize PCoA plots 
hv_plant_niches_all_its<-ggarrange(pcoa_its_all_niche_plot,nrow=1,ncol=1,labels=c("A"),common.legend = TRUE,legend="left", font.label = list(size = 15))

hv_individual_niches_its<-ggarrange(pcoa_its_flower_plot,pcoa_its_leaf_plot,pcoa_its_rhizome_plot,pcoa_its_root_plot, common.legend = T, nrow=2,ncol=2,legend="none", labels=c("B","C","D","E"),font.label = list(size = 15))

plant_plot_its<-ggarrange(hv_plant_niches_all_its, hv_individual_niches_its, widths=c(1,1))


ggsave("beta_diversity_plant_plot_its.png", plot = plant_plot_16s, bg="white", dpi = 300, width = 15, height = 8)


# Soil

# Generating PCoA and ordination plot
pcoa_its_soil<-pcoa_imager(table = met.rareasv.its_soil,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_its_soil_plot<-pcoa_its_soil[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_its_soil_plot

# Conducting PERMANOVA
permanova_its_soil<-as_tibble(adonis2(select_if(met.rareasv.its_soil,is.numeric)~met.rareasv.its_soil$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_its_soil





```


### Beta Dispersion


```{r}
# All Niches

# Calculating sample distances using the vegdist function.  
distances_its_plant<-vegdist(select_if(met.rareasv.its_plant,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_plant<-betadisper(distances_its_plant,met.rareasv.its_plant$Site)
anova(beta_disper_model_its_site_plant)

# Using betadisper function to check for significant variation
beta_disper_model_its_site_niche<-betadisper(distances_its_plant,met.rareasv.its_plant$Tissue_Type)
anova(beta_disper_model_its_site_niche)

# Flower Buds

# Calculating sample distances using the vegdist function.  
distances_its_flower<-vegdist(select_if(met.rareasv.its_flower,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_flower<-betadisper(distances_its_flower,met.rareasv.its_flower$Site)
anova(beta_disper_model_its_site_flower)

# Leaf

# Calculating sample distances using the vegdist function.  
distances_its_leaf<-vegdist(select_if(met.rareasv.its_leaf,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_leaf<-betadisper(distances_its_leaf,met.rareasv.its_leaf$Site)
anova(beta_disper_model_its_site_leaf)

#Rhizomes

# Calculating sample distances using the vegdist function.  
distances_its_rhizome<-vegdist(select_if(met.rareasv.its_rhizome,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_rhizome<-betadisper(distances_its_rhizome,met.rareasv.its_rhizome$Site)
anova(beta_disper_model_its_site_rhizome)

# Roots

# Calculating sample distances using the vegdist function.  
distances_its_root<-vegdist(select_if(met.rareasv.its_root,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_root<-betadisper(distances_its_root,met.rareasv.its_root$Site)
anova(beta_disper_model_its_site_root)


# Soil

# Calculating sample distances using the vegdist function.  
distances_its_soil<-vegdist(select_if(met.rareasv.its_soil,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_its_site_soil<-betadisper(distances_its_soil,met.rareasv.its_soil$Site)
anova(beta_disper_model_its_site_soil)
```



## Bacteria

```{r}
set.seed(123)
# All Niches

# Generating PCoA and ordination plot
pcoa_16s_all_niche<-pcoa_imager(table = met.rareasv.16s_plant,method = "bray",point_ellipse_color = "Tissue_Type",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_all_niche_plot<-pcoa_16s_all_niche[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_all_niche_plot

# Conducting PERMANOVA
permanova_16s_all_niche<-as_tibble(adonis2(select_if(met.rareasv.16s_plant,is.numeric)~met.rareasv.16s_plant$Site*met.rareasv.16s_plant$Tissue_Type,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_all_niche

# Flower Buds

# Generating PCoA and ordination plot
pcoa_16s_leaf<-pcoa_imager(table = met.rareasv.16s_flower,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_flower_plot<-pcoa_16s_flower[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_flower_plot

# Conducting PERMANOVA
permanova_16s_flower<-as_tibble(adonis2(select_if(met.rareasv.16s_flower,is.numeric)~met.rareasv.16s_flower$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_flower


# Leaf

# Generating PCoA and ordination plot
pcoa_16s_leaf<-pcoa_imager(table = met.rareasv.its_leaf,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_leaf_plot<-pcoa_16s_leaf[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_leaf_plot

# Conducting PERMANOVA
permanova_16s_leaf<-as_tibble(adonis2(select_if(met.rareasv.its_leaf,is.numeric)~met.rareasv.its_leaf$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_leaf


# Rhizome

# Generating PCoA and ordination plot
pcoa_16s_rhizome<-pcoa_imager(table = met.rareasv.16s_rhizome,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_rhizome_plot<-pcoa_16s_rhizome[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_rhizome_plot

# Conducting PERMANOVA
permanova_16s_rhizome<-as_tibble(adonis2(select_if(met.rareasv.16s_rhizome,is.numeric)~met.rareasv.16s_rhizome$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_rhizome

# Roots 

# Generating PCoA and ordination plot
pcoa_16s_root<-pcoa_imager(table = met.rareasv.16s_root,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_root_plot<-pcoa_16s_root[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_root_plot

# Conducting PERMANOVA
permanova_16s_root<-as_tibble(adonis2(select_if(met.rareasv.16s_root,is.numeric)~met.rareasv.16s_root$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_root

hv_plant_niches_all_16s<-ggarrange(pcoa_16s_all_niche_plot,nrow=1,ncol=1,labels=c("A"),common.legend = TRUE,legend="left", font.label = list(size = 15))

hv_individual_niches_16s<-ggarrange(pcoa_16s_flower_plot,pcoa_16s_leaf_plot,pcoa_16s_rhizome_plot,pcoa_16s_root_plot, common.legend = T, nrow=2,ncol=2,legend="none", labels=c("B","C","D","E"),font.label = list(size = 15))

plant_plot_16s<-ggarrange(hv_plant_niches_all_16s, hv_individual_niches_16s, widths=c(1,1))

ggsave("beta_diversity_plant_plot_16s.png", plot = plant_plot_16s, bg="white", dpi = 300, width = 15, height = 8)

# Soil

# Generating PCoA and ordination plot
pcoa_16s_soil<-pcoa_imager(table = met.rareasv.16s_soil,method = "bray",point_ellipse_color = "Site",shape = "Site",polygon = TRUE,ellipse_fill = "Site")
pcoa_16s_soil_plot<-pcoa_16s_soil[2][[1]]
  #scale_fill_discrete(labels=c("Autoclave (A)","Non-Autoclave (NA)"))+
  #guides(fill=guide_legend("Soil Amendment"),shape=guide_legend("Septoria Inoculant x Soil Amendment"), color=guide_legend("Septoria Inoculant x Soil Amendment"))
pcoa_16s_soil_plot

# Conducting PERMANOVA
permanova_16s_soil<-as_tibble(adonis2(select_if(met.rareasv.16s_soil,is.numeric)~met.rareasv.16s_soil$Site,method = "bray",permutations = 9999,by="terms"),rownames = "Variable")

permanova_16s_soil
```

### Beta Dispersion

```{r}
# All Niches

# Calculating sample distances using the vegdist function.  
distances_16s_plant<-vegdist(select_if(met.rareasv.16s_plant,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_plant<-betadisper(distances_16s_plant,met.rareasv.16s_plant$Site)
anova(beta_disper_model_16s_site_plant)

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_niche<-betadisper(distances_16s_plant,met.rareasv.16s_plant$Tissue_Type)
anova(beta_disper_model_16s_site_niche)

# Flower Buds

# Calculating sample distances using the vegdist function.  
distances_16s_flower<-vegdist(select_if(met.rareasv.16s_flower,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_flower<-betadisper(distances_16s_flower,met.rareasv.16s_flower$Site)
anova(beta_disper_model_16s_site_flower)

# Leaf

# Calculating sample distances using the vegdist function.  
distances_16s_leaf<-vegdist(select_if(met.rareasv.16s_leaf,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_leaf<-betadisper(distances_16s_leaf,met.rareasv.16s_leaf$Site)
anova(beta_disper_model_16s_site_leaf)

#Rhizomes

# Calculating sample distances using the vegdist function.  
distances_16s_rhizome<-vegdist(select_if(met.rareasv.16s_rhizome,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_rhizome<-betadisper(distances_16s_rhizome,met.rareasv.16s_rhizome$Site)
anova(beta_disper_model_16s_site_rhizome)

# Roots

# Calculating sample distances using the vegdist function.  
distances_16s_root<-vegdist(select_if(met.rareasv.16s_root,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_root<-betadisper(distances_16s_root,met.rareasv.16s_root$Site)
anova(beta_disper_model_16s_site_root)

# Soil

# Calculating sample distances using the vegdist function.  
distances_16s_soil<-vegdist(select_if(met.rareasv.16s_soil,is.numeric), method="bray")

# Using betadisper function to check for significant variation
beta_disper_model_16s_site_soil<-betadisper(distances_16s_soil,met.rareasv.16s_soil$Site)
anova(beta_disper_model_16s_site_soil)

```


# Relative Abundance Charts

## Fungal Phylum Level

### Flower Buds ITS

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_fb_its<-taxonomy_formatting(rare.asvtab =rareasv_its_flower,taxonomy_nochim_its_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_fb_its_phylum<-hv_taxonomy_rare_fb_its[,c(2:19,21)]

# Cleaning taxonomy string
hv_taxonomy_rare_fb_its_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_fb_its_phylum$Phylum)

hv_taxonomy_rare_fb_its_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_fb_its_phylum$Phylum)

hv_taxonomy_rare_fb_its_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_fb_its_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_fb_its_phy_sums<-ddply(hv_taxonomy_rare_fb_its_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_fb_its_phy_sums)<-hv_taxonomy_rare_fb_its_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_fb_its_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_fb_its_phy_sums_meta<-merge(met_helianthus_its,as.data.frame(t(hv_taxonomy_rare_fb_its_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_fb_its_phy_site_sums<-ddply(hv_taxonomy_rare_fb_its_phy_sums_meta[,3:length(hv_taxonomy_rare_fb_its_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_fb_its_phy_site_sums)<-hv_taxonomy_rare_fb_its_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_fb_its_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_fb_its_phy_site_sums_other<-data.frame(hv_taxonomy_rare_fb_its_phy_site_sums[,colSums(hv_taxonomy_rare_fb_its_phy_site_sums)/sum(hv_taxonomy_rare_fb_its_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_fb_its_phy_site_sums[,colSums(hv_taxonomy_rare_fb_its_phy_site_sums)/sum(hv_taxonomy_rare_fb_its_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_fb_its_phy_site_relabund<-decostand(hv_taxonomy_rare_fb_its_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_fb_its_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_fb_its_phy_site_relabund,"Site")
hv_taxonomy_rare_fb_its_phy_site_relabund_long<-gather(hv_taxonomy_rare_fb_its_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_fb_its_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.its.fb<-ggplot(hv_taxonomy_rare_fb_its_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
    scale_fill_manual(values = c( "#F9948E","#700696","#17becf", "#0e334d","#E8B784", "#abd2ed", "#ACD000", "#e377c2","#E8B784")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.its.fb
```

### Leaf ITS

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_lf_its<-taxonomy_formatting(rare.asvtab =rareasv_its_leaf,taxonomy_nochim_its_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_lf_its_phylum<-hv_taxonomy_rare_lf_its[,c(2:22,24)]

# Cleaning taxonomy string
hv_taxonomy_rare_lf_its_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_lf_its_phylum$Phylum)

hv_taxonomy_rare_lf_its_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_lf_its_phylum$Phylum)

hv_taxonomy_rare_lf_its_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_lf_its_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_lf_its_phy_sums<-ddply(hv_taxonomy_rare_lf_its_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_lf_its_phy_sums)<-hv_taxonomy_rare_lf_its_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_lf_its_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_lf_its_phy_sums_meta<-merge(met_helianthus_its,as.data.frame(t(hv_taxonomy_rare_lf_its_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_lf_its_phy_site_sums<-ddply(hv_taxonomy_rare_lf_its_phy_sums_meta[,3:length(hv_taxonomy_rare_lf_its_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_lf_its_phy_site_sums)<-hv_taxonomy_rare_lf_its_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_lf_its_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_lf_its_phy_site_sums_other<-data.frame(hv_taxonomy_rare_lf_its_phy_site_sums[,colSums(hv_taxonomy_rare_lf_its_phy_site_sums)/sum(hv_taxonomy_rare_lf_its_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_lf_its_phy_site_sums[,colSums(hv_taxonomy_rare_lf_its_phy_site_sums)/sum(hv_taxonomy_rare_lf_its_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_lf_its_phy_site_relabund<-decostand(hv_taxonomy_rare_lf_its_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_lf_its_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_lf_its_phy_site_relabund,"Site")
hv_taxonomy_rare_lf_its_phy_site_relabund_long<-gather(hv_taxonomy_rare_lf_its_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_lf_its_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.its.lf<-ggplot(hv_taxonomy_rare_lf_its_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
scale_fill_manual(values = c( "#F9948E","#700696", "#0e334d","#E8B784")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.its.lf
```

### Roots ITS

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_rh_its<-taxonomy_formatting(rare.asvtab =rareasv_its_root,taxonomy_nochim_its_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_rh_its_phylum<-hv_taxonomy_rare_rh_its[,c(2:24,26)]

# Cleaning taxonomy string
hv_taxonomy_rare_rh_its_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_rh_its_phylum$Phylum)

hv_taxonomy_rare_rh_its_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_rh_its_phylum$Phylum)

hv_taxonomy_rare_rh_its_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_rh_its_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_rh_its_phy_sums<-ddply(hv_taxonomy_rare_rh_its_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_rh_its_phy_sums)<-hv_taxonomy_rare_rh_its_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_rh_its_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_rh_its_phy_sums_meta<-merge(met_helianthus_its,as.data.frame(t(hv_taxonomy_rare_rh_its_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_rh_its_phy_site_sums<-ddply(hv_taxonomy_rare_rh_its_phy_sums_meta[,3:length(hv_taxonomy_rare_rh_its_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_rh_its_phy_site_sums)<-hv_taxonomy_rare_rh_its_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_rh_its_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_rh_its_phy_site_sums_other<-data.frame(hv_taxonomy_rare_rh_its_phy_site_sums[,colSums(hv_taxonomy_rare_rh_its_phy_site_sums)/sum(hv_taxonomy_rare_rh_its_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_rh_its_phy_site_sums[,colSums(hv_taxonomy_rare_rh_its_phy_site_sums)/sum(hv_taxonomy_rare_rh_its_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_rh_its_phy_site_relabund<-decostand(hv_taxonomy_rare_rh_its_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_rh_its_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_rh_its_phy_site_relabund,"Site")
hv_taxonomy_rare_rh_its_phy_site_relabund_long<-gather(hv_taxonomy_rare_rh_its_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_rh_its_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.its.rh<-ggplot(hv_taxonomy_rare_rh_its_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
scale_fill_manual(values = c( "#F9948E","#700696", "#0e334d", "#abd2ed", "#e377c2","#E8B784")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.its.rh
```

### Rhizomes ITS

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_rz_its<-taxonomy_formatting(rare.asvtab =rareasv_its_rhizome,taxonomy_nochim_its_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_rz_its_phylum<-hv_taxonomy_rare_rz_its[,c(2:20,22)]

# Cleaning taxonomy string
hv_taxonomy_rare_rz_its_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_rz_its_phylum$Phylum)

hv_taxonomy_rare_rz_its_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_rz_its_phylum$Phylum)

hv_taxonomy_rare_rz_its_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_rz_its_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_rz_its_phy_sums<-ddply(hv_taxonomy_rare_rz_its_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_rz_its_phy_sums)<-hv_taxonomy_rare_rz_its_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_rz_its_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_rz_its_phy_sums_meta<-merge(met_helianthus_its,as.data.frame(t(hv_taxonomy_rare_rz_its_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_rz_its_phy_site_sums<-ddply(hv_taxonomy_rare_rz_its_phy_sums_meta[,3:length(hv_taxonomy_rare_rz_its_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_rz_its_phy_site_sums)<-hv_taxonomy_rare_rz_its_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_rz_its_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_rz_its_phy_site_sums_other<-data.frame(hv_taxonomy_rare_rz_its_phy_site_sums[,colSums(hv_taxonomy_rare_rz_its_phy_site_sums)/sum(hv_taxonomy_rare_rz_its_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_rz_its_phy_site_sums[,colSums(hv_taxonomy_rare_rz_its_phy_site_sums)/sum(hv_taxonomy_rare_rz_its_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_rz_its_phy_site_relabund<-decostand(hv_taxonomy_rare_rz_its_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_rz_its_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_rz_its_phy_site_relabund,"Site")
hv_taxonomy_rare_rz_its_phy_site_relabund_long<-gather(hv_taxonomy_rare_rz_its_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_rz_its_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.its.rz<-ggplot(hv_taxonomy_rare_rz_its_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F9948E","#700696","#17becf", "#0e334d", "#abd2ed", "#ACD000", "#e377c2","#7f7f7f","#E8B784")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.its.rz
```

### Soil ITS

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_sl_its<-taxonomy_formatting(rare.asvtab =rareasv_its_soil,taxonomy_nochim_its_filt_no_negative_soil)

# Subsetting to only include phylum
hv_taxonomy_rare_sl_its_phylum<-hv_taxonomy_rare_sl_its[,c(2:25,27)]

# Cleaning taxonomy string
hv_taxonomy_rare_sl_its_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_sl_its_phylum$Phylum)

hv_taxonomy_rare_sl_its_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_sl_its_phylum$Phylum)

hv_taxonomy_rare_sl_its_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_sl_its_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_sl_its_phy_sums<-ddply(hv_taxonomy_rare_sl_its_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_sl_its_phy_sums)<-hv_taxonomy_rare_sl_its_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_sl_its_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_sl_its_phy_sums_meta<-merge(met_helianthus_its,as.data.frame(t(hv_taxonomy_rare_sl_its_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_sl_its_phy_site_sums<-ddply(hv_taxonomy_rare_sl_its_phy_sums_meta[,3:length(hv_taxonomy_rare_sl_its_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_sl_its_phy_site_sums)<-hv_taxonomy_rare_sl_its_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_sl_its_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_sl_its_phy_site_sums_other<-data.frame(hv_taxonomy_rare_sl_its_phy_site_sums[,colSums(hv_taxonomy_rare_sl_its_phy_site_sums)/sum(hv_taxonomy_rare_sl_its_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_sl_its_phy_site_sums[,colSums(hv_taxonomy_rare_sl_its_phy_site_sums)/sum(hv_taxonomy_rare_sl_its_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_sl_its_phy_site_relabund<-decostand(hv_taxonomy_rare_sl_its_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_sl_its_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_sl_its_phy_site_relabund,"Site")
hv_taxonomy_rare_sl_its_phy_site_relabund_long<-gather(hv_taxonomy_rare_sl_its_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_sl_its_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.its.sl<-ggplot(hv_taxonomy_rare_sl_its_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F9948E","#700696","#17becf", "#0e334d", "#ACD000", "#e377c2", "#7f7f7f","#E8B784")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.its.sl

# Organize Relative Abundance Charts

#1300/800
hv_phyla_niches_its_aboveground<-ggarrange(hv_taxrel.asv.its.fb, hv_taxrel.asv.its.lf,
common.legend = F, nrow=1,ncol=2, labels=c("A","B"),font.label = list(size = 15))


ggsave("hv_phyla_niches_its_aboveground.png", plot = hv_phyla_niches_its_aboveground, dpi = 300, width = 10, height = 5)  

#1300/900
hv_phyla_niches_its_belowground<-ggarrange(hv_taxrel.asv.its.rz,hv_taxrel.asv.its.rh,hv_taxrel.asv.its.sl,
common.legend = F, nrow=2,ncol=2,labels=c("A","B","C"),font.label = list(size = 15))
```

## Bacterial Phylum Level

### Flower Buds 16s

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_fb_16s<-taxonomy_formatting(rare.asvtab =rareasv_16s_flower,taxonomy_nochim_16s_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_fb_16s_phylum<-hv_taxonomy_rare_fb_16s[,c(2:11,13)]

# Cleaning taxonomy string
hv_taxonomy_rare_fb_16s_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_fb_16s_phylum$Phylum)

hv_taxonomy_rare_fb_16s_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_fb_16s_phylum$Phylum)

hv_taxonomy_rare_fb_16s_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_fb_16s_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_fb_16s_phy_sums<-ddply(hv_taxonomy_rare_fb_16s_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_fb_16s_phy_sums)<-hv_taxonomy_rare_fb_16s_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_fb_16s_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_fb_16s_phy_sums_meta<-merge(met_helianthus_16s,as.data.frame(t(hv_taxonomy_rare_fb_16s_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_fb_16s_phy_site_sums<-ddply(hv_taxonomy_rare_fb_16s_phy_sums_meta[,3:length(hv_taxonomy_rare_fb_16s_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_fb_16s_phy_site_sums)<-hv_taxonomy_rare_fb_16s_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_fb_16s_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_fb_16s_phy_site_sums_other<-data.frame(hv_taxonomy_rare_fb_16s_phy_site_sums[,colSums(hv_taxonomy_rare_fb_16s_phy_site_sums)/sum(hv_taxonomy_rare_fb_16s_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_fb_16s_phy_site_sums[,colSums(hv_taxonomy_rare_fb_16s_phy_site_sums)/sum(hv_taxonomy_rare_fb_16s_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_fb_16s_phy_site_relabund<-decostand(hv_taxonomy_rare_fb_16s_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_fb_16s_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_fb_16s_phy_site_relabund,"Site")
hv_taxonomy_rare_fb_16s_phy_site_relabund_long<-gather(hv_taxonomy_rare_fb_16s_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_fb_16s_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.16s.fb<-ggplot(hv_taxonomy_rare_fb_16s_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F5EF1D","#ACD000", "#17becf","#0e334d", "#abd2ed","#ff7f0e","#9467bd", "#F9948E", "#2ca02c", "#e377c2", "#7f7f7f","#8c564b")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.16s.fb
```

### Leaf 16s

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_lf_16s<-taxonomy_formatting(rare.asvtab =rareasv_16s_leaf,taxonomy_nochim_16s_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_lf_16s_phylum<-hv_taxonomy_rare_lf_16s[,c(2:19,21)]

# Cleaning taxonomy string
hv_taxonomy_rare_lf_16s_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_lf_16s_phylum$Phylum)

hv_taxonomy_rare_lf_16s_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_lf_16s_phylum$Phylum)

hv_taxonomy_rare_lf_16s_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_lf_16s_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_lf_16s_phy_sums<-ddply(hv_taxonomy_rare_lf_16s_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_lf_16s_phy_sums)<-hv_taxonomy_rare_lf_16s_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_lf_16s_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_lf_16s_phy_sums_meta<-merge(met_helianthus_16s,as.data.frame(t(hv_taxonomy_rare_lf_16s_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_lf_16s_phy_site_sums<-ddply(hv_taxonomy_rare_lf_16s_phy_sums_meta[,3:length(hv_taxonomy_rare_lf_16s_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_lf_16s_phy_site_sums)<-hv_taxonomy_rare_lf_16s_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_lf_16s_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_lf_16s_phy_site_sums_other<-data.frame(hv_taxonomy_rare_lf_16s_phy_site_sums[,colSums(hv_taxonomy_rare_lf_16s_phy_site_sums)/sum(hv_taxonomy_rare_lf_16s_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_lf_16s_phy_site_sums[,colSums(hv_taxonomy_rare_lf_16s_phy_site_sums)/sum(hv_taxonomy_rare_lf_16s_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_lf_16s_phy_site_relabund<-decostand(hv_taxonomy_rare_lf_16s_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_lf_16s_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_lf_16s_phy_site_relabund,"Site")
hv_taxonomy_rare_lf_16s_phy_site_relabund_long<-gather(hv_taxonomy_rare_lf_16s_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_lf_16s_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.16s.lf<-ggplot(hv_taxonomy_rare_lf_16s_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F5EF1D","#ACD000", "#17becf","#0e334d", "#abd2ed","#ff7f0e","#9467bd", "#F9948E", "#2ca02c", "#e377c2", "#7f7f7f","#8c564b")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.16s.lf
```

### Roots 16s

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_rh_16s<-taxonomy_formatting(rare.asvtab =rareasv_16s_root,taxonomy_nochim_16s_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_rh_16s_phylum<-hv_taxonomy_rare_rh_16s[,c(2:27,29)]

# Cleaning taxonomy string
hv_taxonomy_rare_rh_16s_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_rh_16s_phylum$Phylum)

hv_taxonomy_rare_rh_16s_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_rh_16s_phylum$Phylum)

hv_taxonomy_rare_rh_16s_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_rh_16s_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_rh_16s_phy_sums<-ddply(hv_taxonomy_rare_rh_16s_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_rh_16s_phy_sums)<-hv_taxonomy_rare_rh_16s_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_rh_16s_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_rh_16s_phy_sums_meta<-merge(met_helianthus_16s,as.data.frame(t(hv_taxonomy_rare_rh_16s_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_rh_16s_phy_site_sums<-ddply(hv_taxonomy_rare_rh_16s_phy_sums_meta[,3:length(hv_taxonomy_rare_rh_16s_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_rh_16s_phy_site_sums)<-hv_taxonomy_rare_rh_16s_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_rh_16s_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_rh_16s_phy_site_sums_other<-data.frame(hv_taxonomy_rare_rh_16s_phy_site_sums[,colSums(hv_taxonomy_rare_rh_16s_phy_site_sums)/sum(hv_taxonomy_rare_rh_16s_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_rh_16s_phy_site_sums[,colSums(hv_taxonomy_rare_rh_16s_phy_site_sums)/sum(hv_taxonomy_rare_rh_16s_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_rh_16s_phy_site_relabund<-decostand(hv_taxonomy_rare_rh_16s_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_rh_16s_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_rh_16s_phy_site_relabund,"Site")
hv_taxonomy_rare_rh_16s_phy_site_relabund_long<-gather(hv_taxonomy_rare_rh_16s_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_rh_16s_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.16s.rh<-ggplot(hv_taxonomy_rare_rh_16s_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F5EF1D","#ACD000", "#17becf","#0e334d", "#abd2ed","#ff7f0e","#9467bd", "#F9948E", "#2ca02c", "#e377c2", "#7f7f7f","#8c564b"))+
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.16s.rh
```

### Rhizomes 16s

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_rz_16s<-taxonomy_formatting(rare.asvtab =rareasv_16s_rhizome,taxonomy_nochim_16s_filt_no_negative_plant)

# Subsetting to only include phylum
hv_taxonomy_rare_rz_16s_phylum<-hv_taxonomy_rare_rz_16s[,c(2:23,25)]

# Cleaning taxonomy string
hv_taxonomy_rare_rz_16s_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_rz_16s_phylum$Phylum)

hv_taxonomy_rare_rz_16s_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_rz_16s_phylum$Phylum)

hv_taxonomy_rare_rz_16s_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_rz_16s_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_rz_16s_phy_sums<-ddply(hv_taxonomy_rare_rz_16s_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_rz_16s_phy_sums)<-hv_taxonomy_rare_rz_16s_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_rz_16s_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_rz_16s_phy_sums_meta<-merge(met_helianthus_16s,as.data.frame(t(hv_taxonomy_rare_rz_16s_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_rz_16s_phy_site_sums<-ddply(hv_taxonomy_rare_rz_16s_phy_sums_meta[,3:length(hv_taxonomy_rare_rz_16s_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_rz_16s_phy_site_sums)<-hv_taxonomy_rare_rz_16s_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_rz_16s_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_rz_16s_phy_site_sums_other<-data.frame(hv_taxonomy_rare_rz_16s_phy_site_sums[,colSums(hv_taxonomy_rare_rz_16s_phy_site_sums)/sum(hv_taxonomy_rare_rz_16s_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_rz_16s_phy_site_sums[,colSums(hv_taxonomy_rare_rz_16s_phy_site_sums)/sum(hv_taxonomy_rare_rz_16s_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_rz_16s_phy_site_relabund<-decostand(hv_taxonomy_rare_rz_16s_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_rz_16s_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_rz_16s_phy_site_relabund,"Site")
hv_taxonomy_rare_rz_16s_phy_site_relabund_long<-gather(hv_taxonomy_rare_rz_16s_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_rz_16s_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.16s.rz<-ggplot(hv_taxonomy_rare_rz_16s_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F5EF1D","#ACD000", "#17becf","#0e334d", "#abd2ed","#ff7f0e","#9467bd", "#F9948E", "#2ca02c", "#e377c2", "#7f7f7f","#8c564b")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.16s.rz
```

### Soils 16s

```{r}
# Merging Taxonomy with ASV Table
hv_taxonomy_rare_sl_16s<-taxonomy_formatting(rare.asvtab =rareasv_16s_soil,taxonomy_nochim_16s_filt_no_negative_soil)

# Subsetting to only include phylum
hv_taxonomy_rare_sl_16s_phylum<-hv_taxonomy_rare_sl_16s[,c(2:26,28)]

# Cleaning taxonomy string
hv_taxonomy_rare_sl_16s_phylum$Phylum<-gsub("p__","",hv_taxonomy_rare_sl_16s_phylum$Phylum)

hv_taxonomy_rare_sl_16s_phylum$Phylum<-gsub("_"," ",hv_taxonomy_rare_sl_16s_phylum$Phylum)

hv_taxonomy_rare_sl_16s_phylum$Phylum<-gsub("phy Incertae sedis","Phylum Incertae sedis",hv_taxonomy_rare_sl_16s_phylum$Phylum)

# Calculating col wise sums by phylum
hv_taxonomy_rare_sl_16s_phy_sums<-ddply(hv_taxonomy_rare_sl_16s_phylum,.(Phylum),colwise(sum))

# Making Phyla the Row Names
row.names(hv_taxonomy_rare_sl_16s_phy_sums)<-hv_taxonomy_rare_sl_16s_phy_sums$Phylum

# Removing duplicate column
hv_taxonomy_rare_sl_16s_phy_sums$Phylum<-NULL

# Transposing and merging in metadata
hv_taxonomy_rare_sl_16s_phy_sums_meta<-merge(met_helianthus_16s,as.data.frame(t(hv_taxonomy_rare_sl_16s_phy_sums)),by.x="Sample_ID",by.y=0)

# Calculating col wise sums by site
hv_taxonomy_rare_sl_16s_phy_site_sums<-ddply(hv_taxonomy_rare_sl_16s_phy_sums_meta[,3:length(hv_taxonomy_rare_sl_16s_phy_sums_meta)],.(Site),colwise(sum))

# Changing row names to site name. 
row.names(hv_taxonomy_rare_sl_16s_phy_site_sums)<-hv_taxonomy_rare_sl_16s_phy_site_sums$Site

# Removing duplicate column.
hv_taxonomy_rare_sl_16s_phy_site_sums$Site<-NULL

# Calculating other category (less than 1% of total community)
hv_taxonomy_rare_sl_16s_phy_site_sums_other<-data.frame(hv_taxonomy_rare_sl_16s_phy_site_sums[,colSums(hv_taxonomy_rare_sl_16s_phy_site_sums)/sum(hv_taxonomy_rare_sl_16s_phy_site_sums)>0.01],Other=rowSums(hv_taxonomy_rare_sl_16s_phy_site_sums[,colSums(hv_taxonomy_rare_sl_16s_phy_site_sums)/sum(hv_taxonomy_rare_sl_16s_phy_site_sums)<=0.01]))

# Calculating relative abundance
hv_taxonomy_rare_sl_16s_phy_site_relabund<-decostand(hv_taxonomy_rare_sl_16s_phy_site_sums_other,method = "total")

# Converting to long format
hv_taxonomy_rare_sl_16s_phy_site_relabund<-tibble::rownames_to_column(hv_taxonomy_rare_sl_16s_phy_site_relabund,"Site")
hv_taxonomy_rare_sl_16s_phy_site_relabund_long<-gather(hv_taxonomy_rare_sl_16s_phy_site_relabund,phylum,relative_abundance,2:length(hv_taxonomy_rare_sl_16s_phy_site_relabund),factor_key = TRUE,)

# Plotting bar charts with ggplot
hv_taxrel.asv.16s.sl<-ggplot(hv_taxonomy_rare_sl_16s_phy_site_relabund_long, aes(x=Site, y=relative_abundance, fill=phylum))+
  geom_bar(stat="identity", show.legend=TRUE, color="black")+
  xlab("Site") +
  ylab("Relative Abundance") +
  scale_fill_manual(values = c( "#F5EF1D","#ACD000", "#17becf","#0e334d", "#abd2ed","#ff7f0e","#9467bd", "#2ca02c", "#e377c2", "#7f7f7f","#8c564b")) +
 theme_classic()+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.text.x = element_text(size = 14), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 14))+
    guides(fill=guide_legend(title="Phylum"))+
  scale_y_continuous(breaks=c(0.0,0.25,0.50,0.75,1.0))+
     theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
hv_taxrel.asv.16s.sl

# Organize Relative Abundance Charts
#1300/800
hv_phyla_niches_16s_aboveground<-ggarrange(hv_taxrel.asv.16s.fb, hv_taxrel.asv.16s.lf,
common.legend = F, nrow=1,ncol=2, labels=c("A","B"),font.label = list(size = 15))


ggsave("hv_phyla_niches_16s_aboveground.png", plot = hv_phyla_niches_16s_aboveground, dpi = 300, width = 10, height = 5)  

#1300/900
hv_phyla_niches_16s_belowground<-ggarrange(hv_taxrel.asv.16s.rz,hv_taxrel.asv.16s.rh,hv_taxrel.asv.16s.sl,
common.legend = F, nrow=2,ncol=2,labels=c("A","B","C"),font.label = list(size = 15))
```



# Soil Nutrient Analyses: PCA

```{r}
# Importing soil chemical data from Georgia Soil, Plant and Water Lab
soil_nutrient<-read_excel("Soil_nutrient.xlsx", skip = 8)

#Import metadata
metadata <- read_excel("CultureIndependent_metadata_soilchemistry.xlsx")

metadata <- data.frame(metadata)

# Omitting chemical element that are not of interest 
soil_nutrient<-soil_nutrient[grep("LBC.1",colnames(soil_nutrient),invert=TRUE)]
soil_nutrient<-soil_nutrient[grep("LBCeq",colnames(soil_nutrient),invert=TRUE)]

soil_nutrient<-soil_nutrient[grep("Cr",colnames(soil_nutrient),invert=TRUE)]

soil_nutrient<-soil_nutrient[grep("Base..Satur...ation",colnames(soil_nutrient),invert=TRUE)]

# Removing the _ from sample names to conform to other metadata files. 
soil_nutrient$Sample<-sub("_S_","SL",soil_nutrient$Sample)

soil_nutrient<-data.frame(soil_nutrient)

#Make Sample the the rowname
row.names(soil_nutrient)<-make.names(soil_nutrient$Sample)

#Sanity Check
rownames(soil_nutrient)
colnames(soil_nutrient)

#merge soil_nutrient and metadata by sample

soil_nutrient_metadata <-merge(metadata, soil_nutrient, by.x ="Sample_ID", by.y = "Sample")

#Make Sample_ID the the rowname
row.names(soil_nutrient_metadata)<-make.names(soil_nutrient_metadata$Sample_ID)

# Removing non-numeric data from soil data frame. 
no_numeric_soil_nutrient<-soil_nutrient_metadata[,5:length(soil_nutrient_metadata)]

#Scaling and centering data prior to PCA analysis

scale_soil_nutrient<-scale(no_numeric_soil_nutrient,scale=TRUE,center=TRUE)

# Setting seed to get consistent results. 
set.seed(123)

#Perform PCA using prcomp function of factoextra package. Scale and center are set to false since variables have already been scaled and centered. 
soil_chem_pca<-prcomp(scale_soil_nutrient, scale=FALSE, center=FALSE)

# Using the fviz_eig function to identify eigen values of each principal component. 
fviz_eig(soil_chem_pca)[1]
fviz_pca_biplot(soil_chem_pca)

# Investigating how each soil property loads on the first two principal components. 
#contribution of each variable interms of the variation explained on each axis
#How do soil group in terms of their soil properties
fviz_contrib(soil_chem_pca, choice="var", axes=1, top=10)
fviz_contrib(soil_chem_pca, choice="var", axes=2, top=10)

# Creating a data frame that consists of the first two PCs. 
soil_chem_pca_frame<-data.frame(PC1=scores(soil_chem_pca)[,1],PC2=scores(soil_chem_pca)[,2])

#Sanity Check
rownames(soil_chem_pca_frame)
colnames(soil_chem_pca_frame)


# Merging in metadata that has the sample and site information and nutrient properties with the dataframe containg the first 2 PCs
soil_chem_pca_frame_meta<-merge(soil_nutrient_metadata, soil_chem_pca_frame, by=0)
soil_chem_pca_frame_meta<-data.frame(soil_chem_pca_frame_meta)

#Make Sample_ID the the rowname
row.names(soil_chem_pca_frame_meta)<-make.names(soil_chem_pca_frame_meta$Sample_ID)

#Remove a duplicated column, which is column 1 "Row.names"
soil_chem_pca_frame_meta<-soil_chem_pca_frame_meta[,2:length(soil_chem_pca_frame_meta)]
# Creating a data frame that contains the eigenvectors from the PCA for the soil properties.
soil_pca_eigenvectors<-as.data.frame(soil_chem_pca$rotation[,1:2])

# Creating an object that contains only the eigenvector labels. 
soil_pca_eigenvectors_labels<-labels(soil_pca_eigenvectors)[[1]]

# Determining what factor to use to adjust arrow length for plotting eigenvectors in ggplot. 
soil_pca_arrows_mult<-vegan::ordiArrowMul(scores(soil_chem_pca$rotation[,1:2]))
soil_pca_arrows_mult 

# Generating standard deviation ellipses for PCA by treatment. 

soil_chem_pca_ellipse<-ordiellipse(soil_chem_pca_frame_meta[,24:25],soil_chem_pca_frame_meta$Site, display="sites", kind="sd")

ell_pca_soil_chem <- data.frame()
for(g in levels(as.factor(soil_chem_pca_frame_meta$Site))){
ell_pca_soil_chem <- rbind(ell_pca_soil_chem, cbind(as.data.frame(with(soil_chem_pca_frame_meta[soil_chem_pca$Site==g,],                       vegan:::veganCovEllipse(soil_chem_pca_ellipse[[g]]$cov,soil_chem_pca_ellipse[[g]]$center,soil_chem_pca_ellipse[[g]]$scale))) ,Site=g))}

# Plotting PCA using ggplot and fviz_pca_biplot
soil_pca_biplot<-fviz_pca_biplot(soil_chem_pca,col.ind = soil_nutrient_metadata$Site, col.var = "black",label = "var",invisible = "quali",axes.linetype=NA,geom.ind = FALSE,repel = TRUE)+
  labs(shape="Site",color="Site",fill="Site")+
geom_point(aes(color=soil_nutrient_metadata$Site, shape= soil_nutrient_metadata$Site),size=3)+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  geom_path(data=ell_pca_soil_chem, aes(x=PC1, y=PC2, colour=Site),size=0.5, linetype=1)+
    theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
    geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  guides(fill=guide_legend(title="Site"))+
  xlab("PC1 (44.7%)")+
  ylab("PC2 (17.4%)")+
  ggtitle(NULL)+
  labs(color="Site")+
 theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),
  legend.text=element_text(size=15),legend.title=element_text(size=15))
soil_pca_biplot

# Testing homogeneity of variance using betadisper function

soil_nutrient_metadata
beta_disper_soil_chem_nutrients<-betadisper(vegdist(soil_nutrient_metadata[,5:23],method="bray"),soil_nutrient_metadata$Site)
anova(beta_disper_soil_chem_nutrients)

set.seed(123)
# Testing for significant differences between treatments using adonis for PERMANOVA
adonis2(soil_nutrient_metadata[,5:23]~soil_nutrient_metadata$Site,dist="bray",permutations = 9999,by="terms")

View(soil_nutrient_metadata)

ggsave("soil_nutrient_pca_plot.png", plot = soil_pca_biplot, dpi = 300, width = 10, height = 8)
```


# Soil Nutrient Analyses: Site-Wise Comparisons

```{r}
# Since the data does not meet the assumption of ANOVA after transformation(log and sqrt), a non-parametric approach was used
# N
N_soil_kruskal_helianthus <-kruskal.test(N~Site,soil_nutrient_metadata)
N_soil_kruskal_helianthus 

rstatix::dunn_test(soil_nutrient_metadata,N~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
N_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=N, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("N")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6, aes(x=1,y=0.6,label="AB"))+
  geom_text(data=NULL,size=6, aes(x=2,y=0.6,label="A"))+
  geom_text(data=NULL,size=6, aes(x=3,y=0.6,label="B"))
  
N_soil_plot

#P
P_soil_kruskal_helianthus <-kruskal.test(P~Site,soil_nutrient_metadata)
P_soil_kruskal_helianthus 

rstatix::dunn_test(soil_nutrient_metadata,P~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
P_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=P, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("P (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6, aes(x=1,y=400,label="A"))+
  geom_text(data=NULL,size=6, aes(x=2,y=400,label="B"))+
  geom_text(data=NULL,size=6, aes(x=3,y=400,label="B"))

P_soil_plot

#K
K_soil_kruskal_helianthus<-kruskal.test(K~Site,soil_nutrient_metadata)
K_soil_kruskal_helianthus


#Ca
Ca_soil_anova_helianthus<-kruskal.test(Ca~Site,soil_nutrient_metadata)
Ca_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Ca~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
Ca_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Ca, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Ca (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=10000,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=10000,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=10000,label="B"))
Ca_soil_plot

#Mg
Mg_soil_anova_helianthus<-kruskal.test(Mg~Site,soil_nutrient_metadata)
Mg_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Mg~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
Mg_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Mg, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Mg (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=1100,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=1100,label="B"))+
  geom_text(data=NULL,size=6,aes(x=3,y=1100,label="B"))
Mg_soil_plot



# Organize Macronutrients plots
macronutrient_plots<-ggarrange(N_soil_plot, P_soil_plot, Ca_soil_plot,  Mg_soil_plot, labels=c("A","B","C","D"),legend="right", common.legend=T, ncol = 2, nrow=2, font.label = list(size = 15))


ggsave("macronutrient_plots.png", plot = macronutrient_plots, bg="white", dpi = 300, width = 10, height = 10)  

#Fe
Fe_soil_anova_helianthus<-kruskal.test(Fe~Site,soil_nutrient_metadata)
Fe_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata, Fe~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
Fe_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Fe, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Fe (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6, aes(x=1,y=75,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=75,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=75,label="B"))
Fe_soil_plot

#Mn
Mn_soil_anova_helianthus<-kruskal.test(Mn~Site,soil_nutrient_metadata)
Mn_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata, Mn~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
Mn_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Mn, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Mn (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=200,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=200,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=200,label="B"))
Mn_soil_plot

#Zn
Zn_soil_anova_helianthus<-kruskal.test(Zn~Site,soil_nutrient_metadata)
Zn_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata, Zn~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot.
Zn_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Zn, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Zn (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=75,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=75,label="B"))+
  geom_text(data=NULL,size=6,aes(x=3,y=75,label="A"))
  
Zn_soil_plot

#Mo
Mo_soil_anova_helianthus<-kruskal.test(Mo~Site,soil_nutrient_metadata)
Mo_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata, Mo~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
Mo_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Mo, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Mo (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=0.25,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=0.25,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=0.25,label="B"))
  
Mo_soil_plot


#Ni
Ni_soil_anova_helianthus<-kruskal.test(Ni~Site,soil_nutrient_metadata)
Ni_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Ni~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
Ni_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Ni, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Ni (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=1,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=1,label="B"))+
  geom_text(data=NULL,size=6,aes(x=3,y=1,label="A"))
  
Ni_soil_plot

# Organize Micronutrients plots
micronutrient_plots<- ggarrange(Fe_soil_plot,Mn_soil_plot,Zn_soil_plot,Mo_soil_plot,Ni_soil_plot, labels=c("A","B","C","D","E"), legend="right",common.legend=T, ncol = 2, nrow=3, font.label = list(size = 15))

ggsave("micronutrient_plots.png", plot = micronutrient_plots, bg="white", dpi = 300, width = 10, height = 10)  

#Cd
Cd_soil_anova_helianthus<-kruskal.test(Cd~Site,soil_nutrient_metadata)
Cd_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Cd~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
Cd_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Cd, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Cd (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=0.5,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=0.5,label="B"))+
  geom_text(data=NULL,size=6,aes(x=3,y=0.5,label="AB"))
  
Cd_soil_plot


#Pb
Pb_soil_anova_helianthus<-kruskal.test(Pb~Site,soil_nutrient_metadata)
Pb_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Pb~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
Pb_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Pb, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Pb (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=2,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=2,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=2,label="B"))
  
Pb_soil_plot


# Organize Heavy Metal plots
heavy_metal_plots<-ggarrange(Cd_soil_plot,
          Pb_soil_plot,
          labels=c("A","B"),common.legend=T, ncol = , nrow=1, legend="right",font.label = list(size = 15))

ggsave("heavy_metal_plots.png", plot = heavy_metal_plots, bg="white", dpi = 300, width = 10, height = 8)  


#C
C_soil_anova_helianthus<-kruskal.test(C~Site,soil_nutrient_metadata)
C_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,C~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
C_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=C, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("C %")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=25,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=25,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=25,label="B"))
  
C_soil_plot


#Cu
Cu_soil_anova_helianthus<-kruskal.test(Cu~Site,soil_nutrient_metadata)
Cu_soil_anova_helianthus





#Na
Na_soil_anova_helianthus<-kruskal.test(Na~Site,soil_nutrient_metadata)
Na_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,Na~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
Na_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=Na, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("Na (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=70,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=70,label="A"))+
  geom_text(data=NULL,size=6,aes(x=3,y=70,label="B"))
  
Na_soil_plot
#pH
pH_soil_anova_helianthus<-kruskal.test(pH.2~Site,soil_nutrient_metadata)
pH_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,pH.2~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
pH_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=pH.2, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("pH")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6, aes(x=1,y=10,label="A"))+
  geom_text(data=NULL,size=6, aes(x=2,y=10,label="A"))+
  geom_text(data=NULL,size=6, aes(x=3,y=10,label="B"))
  
pH_soil_plot

#CEC
CEC_soil_anova_helianthus<-kruskal.test(CEC~Site,soil_nutrient_metadata)
CEC_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,CEC~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
CEC_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=CEC, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("CEC (meq/100g)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=80,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=80,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=80,label="B"))
  
CEC_soil_plot

#NH4
NH4_N_soil_anova_helianthus<-kruskal.test(NH4.N~Site,soil_nutrient_metadata)
NH4_N_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,NH4.N~Site)


# Plotting data as a boxplot using ggplot with geom_boxplot. 
NH4_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=NH4.N, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("NH4 (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=35,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=35,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=35,label="B"))
  
NH4_soil_plot

#NO3
NO3_soil_anova_helianthus<-kruskal.test(NO3.N~Site,soil_nutrient_metadata)
NO3_soil_anova_helianthus

rstatix::dunn_test(soil_nutrient_metadata,NO3.N~Site)

# Plotting data as a boxplot using ggplot with geom_boxplot. 
NO3_soil_plot<-ggplot(soil_nutrient_metadata, aes(x=Site, y=NO3.N, outlier.shape=NA,fill=Site))+
  geom_boxplot(outlier.shape = NA)+
  geom_point(position = position_jitterdodge(0.1))+
  xlab("Site")+
  ylab("NO3 (mg/Kg)")+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA),
  axis.text = element_text(face="bold",size=14),
  axis.title = element_text(face="bold",size=14))+
  geom_text(data=NULL,size=6,aes(x=1,y=300,label="A"))+
  geom_text(data=NULL,size=6,aes(x=2,y=300,label="AB"))+
  geom_text(data=NULL,size=6,aes(x=3,y=300,label="B"))
  
NO3_soil_plot

#Organize plots for a measure of soil health indicators 
soil_health_indicators_plots<-ggarrange(C_soil_plot,
          Na_soil_plot,
          pH_soil_plot,
          CEC_soil_plot,
          NH4_soil_plot,
          NO3_soil_plot,
          labels=c("A","B","C","D","E","F"),common.legend=T, ncol = 2, nrow=3, legend="right", font.label = list(size = 15)) 
ggsave("soil_health_indicators_plots.png", plot = soil_health_indicators_plots, bg="white", dpi = 300, width = 10, height = 10)  
```



# Soil Nutrient Analyses: dbRDA

# Fungi
```{r}
#Soil
# Make a Correlation Plot to evaluate the correlation between our current soil variables. 
soil_variables<-data.frame(no_numeric_soil_nutrient)

par(mfrow=c(1,1))

cor_soil_variables<-corrplot(cor(soil_variables,method="pearson"),method="number", tl.srt = 45, tl.cex = 0.8, width = 8, height = 8)

#Remove variables with strong or high correlation with other variables to limit the effect of multi-collinearity in dbRDA

soil_variables_sub1<-data.frame(pH=soil_variables$pH.2,N=soil_variables$N,C=soil_variables$C, Fe=soil_variables$Fe, P=soil_variables$P,Ca=soil_variables$Ca, Mg=soil_variables$Mg,NO3=soil_variables$NO3.N, NH4=soil_variables$NH4.N)



# Generate the corrplot again to evaluate any correlations.

cor_soil_variables_sub<-corrplot(cor(soil_variables_sub1,method="pearson"),method="number")

# since soil properties  that has a strong correlation with  other soil properties. 

soil_variables_sub2<-data.frame(pH=soil_variables$pH.2,N=soil_variables$N,C=soil_nutrient_all$C,P=soil_variables$P,Ca=soil_variables$Ca, Mg=soil_variables$Mg,NO3=soil_variables$NO3.N, NH4=soil_variables$NH4.N)


# Generate the corrplot again to evaluate any correlations.
cor_soil_variables_sub2<-corrplot(cor(soil_variables_sub2,method="pearson"),method="number")


# Creating a new object for soil nutrient properties

soil_nutrient_all<-soil_nutrient_metadata

# Using the selected soil properties 

soil_nutrient_sub<-data.frame(Sample_ID=soil_nutrient_all$Sample_ID, Site=soil_nutrient_all$Site,pH=soil_nutrient_all$pH.2,N=soil_nutrient_all$N,C=soil_nutrient_all$C, P=soil_nutrient_all$P,Ca=soil_nutrient_all$Ca, Mg=soil_nutrient_all$Mg,NO3=soil_nutrient_all$NO3.N, NH4=soil_nutrient_all$NH4.N)

#Make rownames Sample_ID
row.names(soil_nutrient_sub)<-make.names(soil_nutrient_sub$Sample_ID)

# Creat a new object with the metadata rarefied soil asv table  
met.rareasv.its_sl_chem <- met.rareasv.its_soil

# Remove "F" from met.rareasv.its_sl to match sample ID in the selected soil prperties of interest.
met.rareasv.its_sl_chem$Sample_ID <- gsub("F", "", met.rareasv.its_sl_chem$Sample_ID)

# Merging soil ASV table with soil data using sample names.

soil_nutrient_rare_met_its_soils<-merge(soil_nutrient_sub,met.rareasv.its_sl_chem,by.x = 0, by.y="Sample_ID")

#Make rownames Sample_ID
row.names(soil_nutrient_rare_met_its_soils)<-make.names(soil_nutrient_rare_met_its_soils$Sample_ID)
colnames(soil_nutrient_rare_met_its_soils)[3]<-"Site"
colnames(soil_nutrient_rare_met_its_soils)[13]<-"Site"

#Remove duplicated column for Site from soil_nutrient_rare_met_its_soils
soil_nutrient_rare_met_its_soils<-soil_nutrient_rare_met_its_soils[,-13]

# Conducting dbrda with subset soil data
soil_nutrient_sub_dbrda<-vegan::dbrda(soil_nutrient_rare_met_its_soils[,13:length(soil_nutrient_rare_met_its_soils)]~.,soil_nutrient_rare_met_its_soils[,4:11], distance = "bray")

# Creating a dbrda biplot
plot(soil_nutrient_sub_dbrda) 

# Using vif.cca to calculate variance inflation factor of whole model. 
vif_values<-vif.cca(soil_nutrient_sub_dbrda)

#Visualizing VIF values
barplot(vif_values, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)


#Remove C and N from the data set since it inflates the model 

soil_nutrient_rare_met_its_soils_no_C_N <- soil_nutrient_rare_met_its_soils[, !(names(soil_nutrient_rare_met_its_soils) %in% c("C", "N"))]


#Conducting dbRDA using dbrda function
soil_nutrient_sub_dbrda_2<-vegan::dbrda(soil_nutrient_rare_met_its_soils_no_C_N[,11:length(soil_nutrient_rare_met_its_soils_no_C_N)]~.,soil_nutrient_rare_met_its_soils_no_C_N[,4:9], distance = "bray")

# Creating a dbrda biplot
plot(soil_nutrient_sub_dbrda_2)    

# Using vif.cca to calculate variance inflation factor of whole model. 
vif_values_2<-vif.cca(soil_nutrient_sub_dbrda_2)

#Visualizing VIF values
barplot(vif_values_2, main = "VIF Values", horiz = TRUE, col = "steelblue")

# Testing significance of model terms and axes. 
set.seed(123)
anova(soil_nutrient_sub_dbrda_2,by="margin")
anova(soil_nutrient_sub_dbrda_2,by="axis")

                                      
# Generating a data frame that contains only dbrda site scores
soil_nutrient_sub_dbrda_sitescore<-vegan::scores(soil_nutrient_sub_dbrda_2, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
soil_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=soil_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=soil_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_its_helianthus_soil_dbRDA<-met.rareasv.its_soil[,1:3]

# Remove "F" from metadata to match sample ID in the selected soil properties of interest. 
met.rare_its_helianthus_soil_dbRDA$Sample_ID <- gsub("F", "", met.rare_its_helianthus_soil_dbRDA$Sample_ID)

#Make rownames Sample_ID
row.names(met.rare_its_helianthus_soil_dbRDA)<-make.names(met.rare_its_helianthus_soil_dbRDA$Sample_ID)

# Merging first 2 axes score with soil metadata
soil_nutrient_sub_dbrda_graph<-merge(soil_nutrient_sub_dbrda_graph,met.rare_its_helianthus_soil_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(soil_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"


# Extraction eigenvectors and storing them in a data frame.
soil_nutrient_sub_dbrda_arrows<-as.data.frame(soil_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
soil_nutrient_sub_dbrda_arrows_labels<-labels(soil_nutrient_sub_dbrda_arrows)

# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is 0.03173482 I decided to use 4 to ensure the length of arrow is visible
soil_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(soil_nutrient_sub_dbrda_sitescore$biplot))
soil_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
soil_nutrient_sub_dbrda_ellipse<-ordiellipse(soil_nutrient_sub_dbrda_sitescore,soil_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_soil_nutrient <- data.frame()
for(g in levels(as.factor(soil_nutrient_sub_dbrda_graph$Site))){
ell_soil_nutrient <- rbind(ell_soil_nutrient, cbind(as.data.frame(with(soil_nutrient_sub_dbrda_graph[soil_nutrient_sub_dbrda_2$Site==g,],vegan:::veganCovEllipse(soil_nutrient_sub_dbrda_ellipse[[g]]$cov,soil_nutrient_sub_dbrda_ellipse[[g]]$center,soil_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
soil_nutrient_dbrda_all<-ggplot(soil_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_soil_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*soil_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*soil_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*soil_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*soil_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*soil_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*soil_nutrient_sub_arrowmult*1.07,label="Ca"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*soil_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*soil_nutrient_sub_arrowmult*1.07,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*soil_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*soil_nutrient_sub_arrowmult*1.15,label="NO3"),inherit.aes = FALSE,color="black", size=6)+
  
  geom_segment(data=soil_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[6]*soil_nutrient_sub_arrowmult,y=0, yend=dbRDA2[6]*soil_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[6]*soil_nutrient_sub_arrowmult*1.15,y=dbRDA2[6]*soil_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="grey", size=6)+
  geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))

  
soil_nutrient_dbrda_all


#Rhizome

# Creating a new object for soil nutrient properties
rhizomes_nutrient_all_its<-soil_nutrient_metadata


# Using the selected soil properties 

rhizomes_nutrient_sub_its<-data.frame(Sample_ID=rhizomes_nutrient_all_its$Sample_ID,Site=rhizomes_nutrient_all_its$Site,pH=rhizomes_nutrient_all_its$pH.2,P=rhizomes_nutrient_all_its$P,Ca=rhizomes_nutrient_all_its$Ca,Mg=rhizomes_nutrient_all_its$Mg,NO3=rhizomes_nutrient_all_its$NO3.N, NH4=rhizomes_nutrient_all_its$NH4.N)

#Make rownames the Sample_ID

row.names(rhizomes_nutrient_sub_its)<-make.names(rhizomes_nutrient_sub_its$Sample_ID)

# Creat a new object to store soil asv table  
met.rareasv.its_rhizomes_chem <- met.rareasv.its_rhizome


# Reformat sample ID information to match sample ID in the selected soil properties of interest. 
met.rareasv.its_rhizomes_chem$Sample_ID <- gsub("^F", "", met.rareasv.its_rhizomes_chem$Sample_ID)

met.rareasv.its_rhizomes_chem$Sample_ID <- gsub("RZ", "SL", met.rareasv.its_rhizomes_chem$Sample_ID)


# Merging soil ASV table with soil data using sample names.

rhizomes_nutrient_rare_met_its_soils<-merge(rhizomes_nutrient_sub_its,met.rareasv.its_rhizomes_chem,by.x = 0, by.y="Sample_ID")

rownames(rhizomes_nutrient_rare_met_its_soils)

#Make row names to Sample_ID
row.names(rhizomes_nutrient_rare_met_its_soils)<-make.names(rhizomes_nutrient_rare_met_its_soils$Sample_ID)

#Remove duplicated column for and Sample_ID and Site 
rhizomes_nutrient_rare_met_its_soils<-rhizomes_nutrient_rare_met_its_soils[,-1]
rhizomes_nutrient_rare_met_its_soils<-rhizomes_nutrient_rare_met_its_soils[,-10]


#Change Site.x to Site in column 3
colnames(rhizomes_nutrient_rare_met_its_soils)[2]<-"Site"
#Conducting dbRDA using dbrda function
set.seed(123)
rhizomes_nutrient_sub_dbrda_its<-vegan::dbrda(rhizomes_nutrient_rare_met_its_soils[,10:length(rhizomes_nutrient_rare_met_its_soils)]~.,rhizomes_nutrient_rare_met_its_soils[,3:8], distance = "bray")

# Creating a dbrda biplot
plot(rhizomes_nutrient_sub_dbrda_its)    



                                       
# Using vif.cca to calculate variance inflation factor of whole model. 
vif_its_rz<-vif.cca(rhizomes_nutrient_sub_dbrda_its)

#Visualizing VIF values
barplot(vif_its_rz, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)


# Using par to reset figure row displays to 1 x 1. 
par(mfrow=c(1,1))

# Testing significance of model terms and axes. 
set.seed(123)
anova(rhizomes_nutrient_sub_dbrda_its,by="margin")
anova(rhizomes_nutrient_sub_dbrda_its,by="axis")

                                       
# Generating a data frame that contains only dbrda site scores
rhizomes_its_nutrient_sub_dbrda_sitescore<-vegan::scores(rhizomes_nutrient_sub_dbrda_its, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
rhizomes_its_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=rhizomes_its_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=rhizomes_its_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_its_helianthus_rhizomes_dbRDA<-met.rareasv.its_rhizome[1:3]

# Remove "F" from metadata to match sample ID in the selected soil properties of interest. 
met.rare_its_helianthus_rhizomes_dbRDA$Sample_ID <- gsub("^F", "", met.rare_its_helianthus_rhizomes_dbRDA$Sample_ID)

met.rare_its_helianthus_rhizomes_dbRDA$Sample_ID <- gsub("RZ", "SL", met.rare_its_helianthus_rhizomes_dbRDA$Sample_ID)



# Convert rownames to sample ID
row.names(met.rare_its_helianthus_rhizomes_dbRDA)<-make.names(met.rare_its_helianthus_rhizomes_dbRDA$Sample_ID)

# Merging first 2 axes score with soil metadata
rhizomes_its_nutrient_sub_dbrda_graph<-merge(rhizomes_its_nutrient_sub_dbrda_graph,met.rare_its_helianthus_rhizomes_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(rhizomes_its_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"


# Extraction eigenvectors and storing them in a data frame.
rhizomes_its_nutrient_sub_dbrda_arrows<-as.data.frame(rhizomes_its_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
rhizomes_its_nutrient_sub_dbrda_arrows_labels<-labels(rhizomes_its_nutrient_sub_dbrda_arrows)

# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is less tha 1, I decided to use 4 to ensure the length of arrow is visible
rhizomes_its_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(rhizomes_its_nutrient_sub_dbrda_sitescore$biplot))
rhizomes_its_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
rhizomes_its_nutrient_sub_dbrda_ellipse<-ordiellipse(rhizomes_its_nutrient_sub_dbrda_sitescore,rhizomes_its_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_its_rhizomes_nutrient <- data.frame()
for(g in levels(as.factor(rhizomes_its_nutrient_sub_dbrda_graph$Site))){
ell_its_rhizomes_nutrient <- rbind(ell_its_rhizomes_nutrient, cbind(as.data.frame(with(rhizomes_its_nutrient_sub_dbrda_graph[rhizomes_nutrient_sub_dbrda_its$Site==g,],vegan:::veganCovEllipse(rhizomes_its_nutrient_sub_dbrda_ellipse[[g]]$cov,rhizomes_its_nutrient_sub_dbrda_ellipse[[g]]$center,rhizomes_its_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
rhizomes_its_nutrient_dbrda_all<-ggplot(rhizomes_its_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_its_rhizomes_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*rhizomes_its_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*rhizomes_its_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*rhizomes_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*rhizomes_its_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*rhizomes_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*rhizomes_its_nutrient_sub_arrowmult*1.07,label="Ca"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*rhizomes_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*rhizomes_its_nutrient_sub_arrowmult*1.07,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*rhizomes_its_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*rhizomes_its_nutrient_sub_arrowmult*1.15,label="NO3"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[6]*rhizomes_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[6]*rhizomes_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[6]*rhizomes_its_nutrient_sub_arrowmult*1.15,y=dbRDA2[6]*rhizomes_its_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="grey", size=6)+
  
    geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))
rhizomes_its_nutrient_dbrda_all


# Root
# Creating a new object for soil nutrient properties
root_nutrient_all_its<-soil_nutrient_metadata


# Using the selected soil properties 

root_nutrient_sub_its<-data.frame(Sample_ID=root_nutrient_all_its$Sample_ID,Site=root_nutrient_all_its$Site,pH=root_nutrient_all_its$pH.2,P=root_nutrient_all_its$P,Ca=root_nutrient_all_its$Ca, Mg=root_nutrient_all_its$Mg,NO3=root_nutrient_all_its$NO3.N, NH4=root_nutrient_all_its$NH4.N)



#Make rownames Sample_ID
row.names(root_nutrient_sub_its)<-make.names(root_nutrient_sub_its$Sample_ID)

# Creat a new object to store soil asv table  
met.rareasv.its_root_chem <- met.rareasv.its_root


# Reformat sample ID information to match sample ID in the selected soil properties of interest.
met.rareasv.its_root_chem$Sample_ID <- gsub("^F", "", met.rareasv.its_root_chem$Sample_ID)

met.rareasv.its_root_chem$Sample_ID <- gsub("RH", "SL", met.rareasv.its_root_chem$Sample_ID)


# Merging soil ASV table with soil data using sample names.

root_nutrient_rare_met_its_soils<-merge(root_nutrient_sub_its,met.rareasv.its_root_chem,by.x = 0, by.y="Sample_ID")

#Make row names to Sample_ID
row.names(root_nutrient_rare_met_its_soils)<-make.names(root_nutrient_rare_met_its_soils$Sample_ID)

#Remove duplicated column for and Sample_ID and Site 
root_nutrient_rare_met_its_soils<-root_nutrient_rare_met_its_soils[,-1]
root_nutrient_rare_met_its_soils<-root_nutrient_rare_met_its_soils[,-10]


#Change Site.x to Site in column 3
colnames(root_nutrient_rare_met_its_soils)[2]<-"Site"


#Conducting dbRDA using dbrda function
set.seed(123)
root_nutrient_sub_dbrda_its<-vegan::dbrda(root_nutrient_rare_met_its_soils[,10:length(root_nutrient_rare_met_its_soils)]~.,root_nutrient_rare_met_its_soils[,3:8], distance = "bray")

# Creating a dbrda biplot
plot(root_nutrient_sub_dbrda_its)    



                                       
# Using vif.cca to calculate variance inflation factor of whole model. 
vif_its_rt<-vif.cca(root_nutrient_sub_dbrda_its)

#Visualizing VIF values
barplot(vif_its_rt, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)


# Using par to reset figure row displays to 1 x 1. 
par(mfrow=c(1,1))

# Testing significance of model terms and axes. 
set.seed(123)
anova(root_nutrient_sub_dbrda_its,by="margin")
anova(root_nutrient_sub_dbrda_its,by="axis")

                                       
# Generating a data frame that contains only dbrda site scores
root_its_nutrient_sub_dbrda_sitescore<-vegan::scores(root_nutrient_sub_dbrda_its, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
root_its_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=root_its_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=root_its_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_its_helianthus_root_dbRDA<-met.rareasv.its_root[1:3]

# Reformat sample ID information to match sample ID in the selected soil properties of interest. 
met.rare_its_helianthus_root_dbRDA$Sample_ID <- gsub("^F", "", met.rare_its_helianthus_root_dbRDA$Sample_ID)

met.rare_its_helianthus_root_dbRDA$Sample_ID <- gsub("RH", "SL", met.rare_its_helianthus_root_dbRDA$Sample_ID)



# Convert rownames to sample ID
row.names(met.rare_its_helianthus_root_dbRDA)<-make.names(met.rare_its_helianthus_root_dbRDA$Sample_ID)
#Check rownames
rownames(met.rare_its_helianthus_root_dbRDA)

# Merging first 2 axes score with soil metadata
root_its_nutrient_sub_dbrda_graph<-merge(root_its_nutrient_sub_dbrda_graph,met.rare_its_helianthus_root_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(root_its_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"


# Extraction eigenvectors and storing them in a data frame.
root_its_nutrient_sub_dbrda_arrows<-as.data.frame(root_its_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
root_its_nutrient_sub_dbrda_arrows_labels<-labels(root_its_nutrient_sub_dbrda_arrows)


# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is less than 1, I decided to use 4 to ensure the length of arrow is visible
root_its_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(root_its_nutrient_sub_dbrda_sitescore$biplot))
root_its_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
root_its_nutrient_sub_dbrda_ellipse<-ordiellipse(root_its_nutrient_sub_dbrda_sitescore,root_its_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_its_root_nutrient <- data.frame()
for(g in levels(as.factor(root_its_nutrient_sub_dbrda_graph$Site))){
ell_its_root_nutrient <- rbind(ell_its_root_nutrient, cbind(as.data.frame(with(root_its_nutrient_sub_dbrda_graph[root_nutrient_sub_dbrda_its$Site==g,],vegan:::veganCovEllipse(root_its_nutrient_sub_dbrda_ellipse[[g]]$cov,root_its_nutrient_sub_dbrda_ellipse[[g]]$center,root_its_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
root_its_nutrient_dbrda_all<-ggplot(root_its_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_its_root_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*root_its_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*root_its_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*root_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*root_its_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*root_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*root_its_nutrient_sub_arrowmult*1.07,label="Ca"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*root_its_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*root_its_nutrient_sub_arrowmult*1.07,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*root_its_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*root_its_nutrient_sub_arrowmult*1.15,label="NO3"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=root_its_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[6]*root_its_nutrient_sub_arrowmult,y=0, yend=dbRDA2[6]*root_its_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_its_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[6]*root_its_nutrient_sub_arrowmult*1.15,y=dbRDA2[6]*root_its_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))

  
root_its_nutrient_dbrda_all
```

#Bacteria

```{r}
# Soil
# Creating a new object for soil nutrient properties
soil_nutrient_all_bac<-soil_nutrient_metadata

# Using the selected soil properties 
soil_nutrient_sub_bac<-data.frame(Sample_ID=soil_nutrient_all_bac$Sample_ID,Site=soil_nutrient_all_bac$Site,pH=soil_nutrient_all_bac$pH.2,N=soil_nutrient_all_bac$N,P=soil_nutrient_all_bac$P,Ca=soil_nutrient_all_bac$Ca, Mg=soil_nutrient_all_bac$Mg,NO3=soil_nutrient_all_bac$NO3.N, NH4=soil_nutrient_all_bac$NH4.N)

# Creat a new object to store soil asv table  
met.rareasv.16s_sl_chem <- met.rareasv.16s_soil


# Reformat sample ID information to match sample ID in the selected soil properties of interest.
met.rareasv.16s_sl_chem$Sample_ID <- gsub("^B", "", met.rareasv.16s_sl_chem$Sample_ID)

# Merging soil ASV table with soil data using sample names.

soil_nutrient_rare_met_16s_soils<-merge(soil_nutrient_sub_bac,met.rareasv.16s_sl_chem,by.x = "Sample_ID", by.y="Sample_ID")


#Make row names to Sample_ID
row.names(soil_nutrient_rare_met_16s_soils)<-make.names(soil_nutrient_rare_met_16s_soils$Sample_ID)

#Remove duplicated column for and Site 
soil_nutrient_rare_met_16s_soils<-soil_nutrient_rare_met_16s_soils[,-11]

#Change Site.x to Site in column 3
colnames(soil_nutrient_rare_met_16s_soils)[2]<-"Site"

#Conducting dbRDA using dbrda function
set.seed(123)
soil_nutrient_sub_dbrda_16s<-vegan::dbrda(soil_nutrient_rare_met_16s_soils[,11:length(soil_nutrient_rare_met_16s_soils)]~.,soil_nutrient_rare_met_16s_soils[,3:9], distance = "bray")

# Creating a dbrda biplot
plot(soil_nutrient_sub_dbrda_16s) 

# Using vif.cca to calculate variance inflation factor of whole model. 
vif_16s<-vif.cca(soil_nutrient_sub_dbrda_16s)

#Visualizing VIF values
barplot(vif_16s, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)

# Testing significance of model terms and axes. 
set.seed(123)
anova(soil_nutrient_sub_dbrda_16s,by="margin")
anova(soil_nutrient_sub_dbrda_16s,by="axis")

                                       
# Generating a data frame that contains only dbrda site scores
soil_16s_nutrient_sub_dbrda_sitescore<-vegan::scores(soil_nutrient_sub_dbrda_16s, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
soil_16s_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=soil_16s_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=soil_16s_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_16s_helianthus_soil_dbRDA<-met.rareasv.16s_soil[,1:3]

# Reformat sample ID information to match sample ID in the selected soil properties of interest.
met.rare_16s_helianthus_soil_dbRDA$Sample_ID <- gsub("^B", "", met.rare_16s_helianthus_soil_dbRDA$Sample_ID)


# Make rownames to sample ID
row.names(met.rare_16s_helianthus_soil_dbRDA)<-make.names(met.rare_16s_helianthus_soil_dbRDA$Sample_ID)

# Merging first 2 axes score with soil metadata
soil_16s_nutrient_sub_dbrda_graph<-merge(soil_16s_nutrient_sub_dbrda_graph,met.rare_16s_helianthus_soil_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(soil_16s_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"

# Extraction eigenvectors and storing them in a data frame.
soil_16s_nutrient_sub_dbrda_arrows<-as.data.frame(soil_16s_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
soil_16s_nutrient_sub_dbrda_arrows_labels<-labels(soil_16s_nutrient_sub_dbrda_arrows)

# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is less than 1, I decided to use 4 to ensure the length of arrow is visible
soil_16s_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(soil_16s_nutrient_sub_dbrda_sitescore$biplot))
soil_16s_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
soil_16s_nutrient_sub_dbrda_ellipse<-ordiellipse(soil_16s_nutrient_sub_dbrda_sitescore,soil_16s_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_16s_soil_nutrient <- data.frame()
for(g in levels(as.factor(soil_16s_nutrient_sub_dbrda_graph$Site))){
ell_16s_soil_nutrient <- rbind(ell_16s_soil_nutrient, cbind(as.data.frame(with(soil_16s_nutrient_sub_dbrda_graph[soil_nutrient_sub_dbrda_16s$Site==g,],vegan:::veganCovEllipse(soil_16s_nutrient_sub_dbrda_ellipse[[g]]$cov,soil_16s_nutrient_sub_dbrda_ellipse[[g]]$center,soil_16s_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
soil_16s_nutrient_dbrda_all<-ggplot(soil_16s_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_16s_soil_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*soil_16s_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*soil_16s_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*soil_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*soil_16s_nutrient_sub_arrowmult*1.07,label="N"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*soil_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*soil_16s_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*soil_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*soil_16s_nutrient_sub_arrowmult*1.07,label="Ca"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*soil_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*soil_16s_nutrient_sub_arrowmult*1.15,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[6]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[6]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[6]*soil_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[6]*soil_16s_nutrient_sub_arrowmult*1.15,label="NO3"),inherit.aes = FALSE,color="grey", size=6)+
  
   geom_segment(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[7]*soil_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[7]*soil_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=soil_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[7]*soil_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[7]*soil_16s_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))

  
soil_16s_nutrient_dbrda_all


#Rhizomes

# Creating a new object for soil nutrient properties
rhizomes_nutrient_all_bac<-soil_nutrient_metadata


# Using the selected soil properties 

rhizomes_nutrient_sub_bac<-data.frame(Sample_ID=rhizomes_nutrient_all_bac$Sample_ID,Site=rhizomes_nutrient_all_bac$Site,pH=rhizomes_nutrient_all_bac$pH.2,P=rhizomes_nutrient_all_bac$P, Mg=rhizomes_nutrient_all_bac$Mg,NO3=rhizomes_nutrient_all_bac$NO3.N, NH4=rhizomes_nutrient_all_bac$NH4.N)


#Make row nanes Sample_ID
row.names(rhizomes_nutrient_sub_bac)<-make.names(rhizomes_nutrient_sub_bac$Sample_ID)

# Creat a new object to store soil asv table  
met.rareasv.16s_rhizomes_chem <- met.rareasv.16s_rhizome


# Reformat sample ID information to match sample ID in the selected soil properties of interest.
met.rareasv.16s_rhizomes_chem$Sample_ID <- gsub("^B", "", met.rareasv.16s_rhizomes_chem$Sample_ID)
met.rareasv.16s_rhizomes_chem$Sample_ID <- gsub("RZ", "SL", met.rareasv.16s_rhizomes_chem$Sample_ID)


# Merging soil ASV table with soil data using sample names.

rhizomes_nutrient_rare_met_16s_soils<-merge(rhizomes_nutrient_sub_bac,met.rareasv.16s_rhizomes_chem,by.x = 0, by.y="Sample_ID")

rownames(rhizomes_nutrient_rare_met_16s_soils)

#Converting row names to sample names
row.names(rhizomes_nutrient_rare_met_16s_soils)<-make.names(rhizomes_nutrient_rare_met_16s_soils$Sample_ID)

#Remove duplicated column for and Sample_ID and Site 
rhizomes_nutrient_rare_met_16s_soils<-rhizomes_nutrient_rare_met_16s_soils[,-1]
rhizomes_nutrient_rare_met_16s_soils<-rhizomes_nutrient_rare_met_16s_soils[,-9]


#Change Site.x to Site in column 3
colnames(rhizomes_nutrient_rare_met_16s_soils)[2]<-"Site"


#Conducting dbRDA using dbrda function
set.seed(123)
rhizomes_nutrient_sub_dbrda_16s<-vegan::dbrda(rhizomes_nutrient_rare_met_16s_soils[,9:length(rhizomes_nutrient_rare_met_16s_soils)]~.,rhizomes_nutrient_rare_met_16s_soils[,3:7], distance = "bray")

# Creating a dbrda biplot
plot(rhizomes_nutrient_sub_dbrda_16s)    

                                       
# Using vif.cca to calculate variance inflation factor of whole model. 
vif_16s_rz<-vif.cca(rhizomes_nutrient_sub_dbrda_16s)

#Visualizing VIF values
barplot(vif_16s_rz, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)


# Using par to reset figure row displays to 1 x 1. 
par(mfrow=c(1,1))

# Testing significance of model terms and axes. 
set.seed(123)
anova(rhizomes_nutrient_sub_dbrda_16s,by="margin")
anova(rhizomes_nutrient_sub_dbrda_16s,by="axis")

                                       
# Generating a data frame that contains only dbrda site scores
rhizomes_16s_nutrient_sub_dbrda_sitescore<-vegan::scores(rhizomes_nutrient_sub_dbrda_16s, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
rhizomes_16s_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=rhizomes_16s_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=rhizomes_16s_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_16s_helianthus_rhizomes_dbRDA<-met.rareasv.16s_rhizome[1:3]

# Reformat sample ID information to match sample ID in the selected soil properties of interest. 
met.rare_16s_helianthus_rhizomes_dbRDA$Sample_ID <- gsub("^B", "", met.rare_16s_helianthus_rhizomes_dbRDA$Sample_ID)

met.rare_16s_helianthus_rhizomes_dbRDA$Sample_ID <- gsub("RZ", "SL", met.rare_16s_helianthus_rhizomes_dbRDA$Sample_ID)



# Convert rownames to sample ID
row.names(met.rare_16s_helianthus_rhizomes_dbRDA)<-make.names(met.rare_16s_helianthus_rhizomes_dbRDA$Sample_ID)
#Check rownames
rownames(met.rare_16s_helianthus_rhizomes_dbRDA)

# Merging first 2 axes score with soil metadata
rhizomes_16s_nutrient_sub_dbrda_graph<-merge(rhizomes_16s_nutrient_sub_dbrda_graph,met.rare_16s_helianthus_rhizomes_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(rhizomes_16s_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"


# Extraction eigenvectors and storing them in a data frame.
rhizomes_16s_nutrient_sub_dbrda_arrows<-as.data.frame(rhizomes_16s_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
rhizomes_16s_nutrient_sub_dbrda_arrows_labels<-labels(rhizomes_16s_nutrient_sub_dbrda_arrows)

# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is less than 1, I decided to use 4 to ensure the length of arrow is visible
rhizomes_16s_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(rhizomes_16s_nutrient_sub_dbrda_sitescore$biplot))
rhizomes_16s_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
rhizomes_16s_nutrient_sub_dbrda_ellipse<-ordiellipse(rhizomes_16s_nutrient_sub_dbrda_sitescore,rhizomes_16s_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_16s_rhizomes_nutrient <- data.frame()
for(g in levels(as.factor(rhizomes_16s_nutrient_sub_dbrda_graph$Site))){
ell_16s_rhizomes_nutrient <- rbind(ell_16s_rhizomes_nutrient, cbind(as.data.frame(with(rhizomes_16s_nutrient_sub_dbrda_graph[rhizomes_nutrient_sub_dbrda_16s$Site==g,],vegan:::veganCovEllipse(rhizomes_16s_nutrient_sub_dbrda_ellipse[[g]]$cov,rhizomes_16s_nutrient_sub_dbrda_ellipse[[g]]$center,rhizomes_16s_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
rhizomes_16s_nutrient_dbrda_all<-ggplot(rhizomes_16s_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_16s_rhizomes_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*rhizomes_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*rhizomes_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*rhizomes_16s_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*rhizomes_16s_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*rhizomes_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*rhizomes_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*rhizomes_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*rhizomes_16s_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*rhizomes_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*rhizomes_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*rhizomes_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*rhizomes_16s_nutrient_sub_arrowmult*1.07,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*rhizomes_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*rhizomes_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*rhizomes_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*rhizomes_16s_nutrient_sub_arrowmult*1.07,label="NO3"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*rhizomes_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*rhizomes_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=rhizomes_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*rhizomes_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*rhizomes_16s_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="black", size=6)+
  
    geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))

rhizomes_16s_nutrient_dbrda_all

#Root

# Creating a new object for soil nutrient properties
root_nutrient_all_bac<-soil_nutrient_metadata


# Using the selected soil properties 

root_nutrient_sub_bac<-data.frame(Sample_ID=root_nutrient_all_bac$Sample_ID,Site=root_nutrient_all_bac$Site,pH=root_nutrient_all_bac$pH.2,N=root_nutrient_all_bac$N,P=root_nutrient_all_bac$P,Ca=root_nutrient_all_bac$Ca, Mg=root_nutrient_all_bac$Mg,NO3=root_nutrient_all_bac$NO3.N, NH4=root_nutrient_all_bac$NH4.N)




#Make Sample_ID the rownames
row.names(root_nutrient_sub_bac)<-make.names(root_nutrient_sub_bac$Sample_ID)

# Creat a new object to store soil asv table  
met.rareasv.16s_root_chem <- met.rareasv.16s_root


# Reformat sample ID information to match sample ID in the selected soil properties of interest.
met.rareasv.16s_root_chem$Sample_ID <- gsub("^B", "", met.rareasv.16s_root_chem$Sample_ID)


met.rareasv.16s_root_chem$Sample_ID <- gsub("RH", "SL", met.rareasv.16s_root_chem$Sample_ID)


# Merging soil ASV table with soil data using sample names.

root_nutrient_rare_met_16s_soils<-merge(root_nutrient_sub_bac,met.rareasv.16s_root_chem,by.x = 0, by.y="Sample_ID")



#Converting row names to sample names
row.names(root_nutrient_rare_met_16s_soils)<-make.names(root_nutrient_rare_met_16s_soils$Sample_ID)

#Remove duplicated column for and Sample_ID and Site 
root_nutrient_rare_met_16s_soils<-root_nutrient_rare_met_16s_soils[,-1]
root_nutrient_rare_met_16s_soils<-root_nutrient_rare_met_16s_soils[,-11]


#Change Site.x to Site in column 3
colnames(root_nutrient_rare_met_16s_soils)[2]<-"Site"


#Conducting dbRDA using dbrda function
set.seed(123)
root_nutrient_sub_dbrda_16s<-vegan::dbrda(root_nutrient_rare_met_16s_soils[,11:length(root_nutrient_rare_met_16s_soils)]~.,root_nutrient_rare_met_16s_soils[,3:9], distance = "bray")

# Creating a dbrda biplot
plot(root_nutrient_sub_dbrda_16s)    



                                       
# Using vif.cca to calculate variance inflation factor of whole model. 
vif_16s_rt<-vif.cca(root_nutrient_sub_dbrda_16s)

#Visualizing VIF values
barplot(vif_16s_rt, main = "VIF Values", horiz = TRUE, col = "steelblue")

#add vertical line at 5
abline(v = 5, lwd = 3, lty = 2)


# Using par to reset figure row displays to 1 x 1. 
par(mfrow=c(1,1))

# Testing significance of model terms and axes. 
set.seed(123)
anova(root_nutrient_sub_dbrda_16s,by="margin")
anova(root_nutrient_sub_dbrda_16s,by="axis")

                                       
# Generating a data frame that contains only dbrda site scores
root_16s_nutrient_sub_dbrda_sitescore<-vegan::scores(root_nutrient_sub_dbrda_16s, display=c("sites","bp"))

# Extraction only first 2 axes of dbrda
root_16s_nutrient_sub_dbrda_graph<-data.frame(dbRDA1=root_16s_nutrient_sub_dbrda_sitescore$sites[,1], dbRDA2=root_16s_nutrient_sub_dbrda_sitescore$sites[,2])

# Creat an object to store the rarefied metadata for soil

met.rare_16s_helianthus_root_dbRDA<-met.rareasv.16s_root[1:3]

# Reformat sample ID information to match sample ID in the selected soil properties of interest. 
met.rare_16s_helianthus_root_dbRDA$Sample_ID <- gsub("^B", "", met.rare_16s_helianthus_root_dbRDA$Sample_ID)

met.rare_16s_helianthus_root_dbRDA$Sample_ID <- gsub("RH", "SL", met.rare_16s_helianthus_root_dbRDA$Sample_ID)



# Convert rownames to sample ID
row.names(met.rare_16s_helianthus_root_dbRDA)<-make.names(met.rare_16s_helianthus_root_dbRDA$Sample_ID)
#Check rownames
rownames(met.rare_16s_helianthus_root_dbRDA)

# Merging first 2 axes score with soil metadata
root_16s_nutrient_sub_dbrda_graph<-merge(root_16s_nutrient_sub_dbrda_graph,met.rare_16s_helianthus_root_dbRDA,by.x=0,by="Sample_ID")

#Rename Row.names to Sample_ID

colnames(root_16s_nutrient_sub_dbrda_graph)[1]<-"Sample_ID"


# Extraction eigenvectors and storing them in a data frame.
root_16s_nutrient_sub_dbrda_arrows<-as.data.frame(root_16s_nutrient_sub_dbrda_sitescore$biplot)

# Extraction eigen vector labels.
root_16s_nutrient_sub_dbrda_arrows_labels<-labels(root_16s_nutrient_sub_dbrda_arrows)

# Determining what factor to multiply eigenvectors by for plotting in ggplot.
#Since the value is less than 1, I decided to use 4 to ensure the length of arrow is visible
root_16s_nutrient_sub_arrowmult<-vegan::ordiArrowMul(scores(root_16s_nutrient_sub_dbrda_sitescore$biplot))
root_16s_nutrient_sub_arrowmult <- 4

# Generating standard deviation elipses by site
root_16s_nutrient_sub_dbrda_ellipse<-ordiellipse(root_16s_nutrient_sub_dbrda_sitescore,root_16s_nutrient_sub_dbrda_graph$Site, display="sites", kind="sd",draw="none")


ell_16s_root_nutrient <- data.frame()
for(g in levels(as.factor(root_16s_nutrient_sub_dbrda_graph$Site))){
ell_16s_root_nutrient <- rbind(ell_16s_root_nutrient, cbind(as.data.frame(with(root_16s_nutrient_sub_dbrda_graph[root_nutrient_sub_dbrda_16s$Site==g,],vegan:::veganCovEllipse(root_16s_nutrient_sub_dbrda_ellipse[[g]]$cov,root_16s_nutrient_sub_dbrda_ellipse[[g]]$center,root_16s_nutrient_sub_dbrda_ellipse[[g]]$scale))) ,Site=g))}
                                                                    
                                                                  

# Plotting dbrda biplot in ggplot
root_16s_nutrient_dbrda_all<-ggplot(root_16s_nutrient_sub_dbrda_graph, aes(dbRDA1,dbRDA2, colour=Site))+
     geom_point(size=3)+
geom_path(data=ell_16s_root_nutrient, aes(x=dbRDA1, y=dbRDA2, colour=Site),size=0.5, linetype=1)+
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
    panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
  panel.border=element_rect(color="black", size=1, fill=NA))+
  
geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[1]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[1]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="black")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[1]*root_16s_nutrient_sub_arrowmult*1.2,y=dbRDA2[1]*root_16s_nutrient_sub_arrowmult*1.2,label="pH"),inherit.aes = FALSE,color="black", size=6)+
  

geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[2]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[2]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[2]*root_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[2]*root_16s_nutrient_sub_arrowmult*1.07,label="N"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[3]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[3]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[3]*root_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[3]*root_16s_nutrient_sub_arrowmult*1.07,label="P"),inherit.aes = FALSE,color="grey", size=6)+
  
geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[4]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[4]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[4]*root_16s_nutrient_sub_arrowmult*1.07,y=dbRDA2[4]*root_16s_nutrient_sub_arrowmult*1.07,label="Ca"),inherit.aes = FALSE,color="grey", size=6)+

geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[5]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[5]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[5]*root_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[5]*root_16s_nutrient_sub_arrowmult*1.15,label="Mg"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[6]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[6]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[6]*root_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[6]*root_16s_nutrient_sub_arrowmult*1.15,label="NO3"),inherit.aes = FALSE,color="grey", size=6)+
  
   geom_segment(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=0, xend=dbRDA1[7]*root_16s_nutrient_sub_arrowmult,y=0, yend=dbRDA2[7]*root_16s_nutrient_sub_arrowmult),inherit.aes = FALSE,arrow =arrow(length = unit(0.25, "cm")), size=0.35,linetype=1, colour="grey")+
  geom_text(data=root_16s_nutrient_sub_dbrda_arrows,aes(x=dbRDA1[7]*root_16s_nutrient_sub_arrowmult*1.15,y=dbRDA2[7]*root_16s_nutrient_sub_arrowmult*1.15,label="NH4"),inherit.aes = FALSE,color="grey", size=6)+
  
  geom_hline(yintercept=0,size=0.02)+
  geom_vline(xintercept=0,size=0.02)+
  theme(axis.text = element_text(face="bold"),
  axis.title = element_text(face="bold"),legend.text=element_text(size=15),legend.title=element_text(size=15))

  
root_16s_nutrient_dbrda_all


#Organize the soil beta diversity plot andsoil dbRDA plot for bacteria and fungi

bac_its_betadiversity_soil<-ggarrange(pcoa_16s_soil_plot, pcoa_its_soil_plot,  common.legend = T, nrow=2, ncol=1,legend="left", labels=c("A","B"),font.label = list(size = 15))

bac_its_dbRDA_soil<-ggarrange(soil_16s_nutrient_dbrda_all, soil_nutrient_dbrda_all,  common.legend = T, nrow=2, ncol=1,legend="right", labels=c("C","D"),font.label = list(size = 15))

soil_betadiversity_dbrda<-ggarrange(bac_its_betadiversity_soil, bac_its_dbRDA_soil, widths=c(1,1), font.label = list(size = 15))
ggsave("soil_betadiversity_dbrda.png", plot = soil_betadiversity_dbrda, bg="white",dpi = 300, width = 12, height = 6) 

#Organize rhizome and root plots for bacteria 
bac_dbRDA_rhizomes_roots<-ggarrange(rhizomes_16s_nutrient_dbrda_all, root_16s_nutrient_dbrda_all,  common.legend = T, nrow=1, ncol=2,legend="right", labels=c("A","B"),font.label = list(size = 15))

ggsave("bac_dbRDA_rhizomes_roots.png", plot = bac_dbRDA_rhizomes_roots, bg="white",dpi = 300, width = 12, height = 8) 


#Organize rhizome and root plots for fungi
its_dbRDA_rhizomes_roots<-ggarrange(rhizomes_its_nutrient_dbrda_all, root_its_nutrient_dbrda_all,  common.legend = T, nrow=1, ncol=2,legend="right", labels=c("A","B"),font.label = list(size = 15))

ggsave("its_dbRDA_rhizomes_roots.png", plot = its_dbRDA_rhizomes_roots, bg="white",dpi = 300, width = 12, height = 8) 
```



